---
title: "RPKM"
output:
  html_document:
    toc: true
    theme: united
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "~/Documentos/Repos/UOC-Master/data_testing/")
```

# Calculate RPKM

## RGI

```{r message=FALSE}
library(readr)
RGI_data <- read_delim("RGI_analysis/argannot-150bp-10000-reads.fastq_RGI_output.allele_mapping_data.txt", "\t", escape_double = FALSE, trim_ws = TRUE)
```


```{r}
library(dplyr)
totalNumReads<-	10000
RGI_data<-RGI_data%>%mutate(RPKM=`All Mapped Reads` /(`Reference Length`/1000 * totalNumReads/1000000))

```

### Average MAPQ distribution

```{r}
library(ggplot2)
ggplot(RGI_data, aes(`Average MAPQ (Completely Mapped Reads)`)) +
  geom_histogram(bins=50, aes(y = stat(density)))+geom_density()+theme_bw()

ggplot(RGI_data, aes(`Average MAPQ (Completely Mapped Reads)`)) +
  geom_histogram(bins=50)+theme_bw()

```

###  Apply Average MAPQ filter:

```{r}
RGI_data<-RGI_data%>%filter(`Average MAPQ (Completely Mapped Reads)`>10)
```

### RPKM sum per AMR gene family

```{r}
library(knitr)
n_distinct(RGI_data$`AMR Gene Family`)
AMR_family_RGI<-RGI_data%>%
  group_by(`AMR Gene Family`)%>%
  summarise('number of AMR genes per family (RGI)'=n(),
            sum_RPKM_RGI=sum(RPKM),
            avg_MAPQ=mean(`Average MAPQ (Completely Mapped Reads)`),
            q1_MAPQ=quantile(`Average MAPQ (Completely Mapped Reads)`, 0.25),
            p3_MAPQ=quantile(`Average MAPQ (Completely Mapped Reads)`,0.75),)%>%arrange(desc(sum_RPKM_RGI))
kable(head(AMR_family_RGI))
str(AMR_family_RGI$`sum RPKM (RGI)`)
```


# ARIBA

```{r message=FALSE}
Ariba_data <- read_delim("ariba_analysis/argannot-150bp-10000-reads_ariba_results/report.tsv", 
    "\t", escape_double = FALSE, trim_ws = TRUE)
```
report1: only coding sequences
report2: all sequences
Remove contig and variant data

```{r}
##Ariba_data<-Ariba_data %>% filter(gene==1)%>%select(c(1:8,31))%>%unique()
Ariba_data<-Ariba_data%>%select(c(1:8,31))%>%unique()
```
I will use the total nÂº of reads instead mapped reads

```{r}
Ariba_data<-Ariba_data%>%mutate(RPKM=reads /(ref_len/1000 * totalNumReads/1000000))
```

### Obtain AMR gene family
Import CARD data aro index: 

```{r message=FALSE}
library(readr)
aro_index <- read_delim("~/Documentos/Repos/UOC-Master/PRJNA307231/aro_index.tsv", 
    "\t", escape_double = FALSE, trim_ws = TRUE)
```

Trasform ARIBA notation to merge with CARD  DB
```{r}
library(stringr)
library(reshape2)
y<-colsplit(Ariba_data$ref_name, "\\.", c("ARO Term", "ARO Acession","NCBI","init_final","Model Sequence ID","X"))
y$NCBI<-ifelse(str_detect(y$init_final, "_"),y$NCBI ,paste(y$NCBI,".",y$init_final) )
y$init_final<-ifelse(str_detect(y$init_final, "_"), y$init_final, y$`Model Sequence ID`)
y$`Model Sequence ID`<-ifelse(str_detect(y$`Model Sequence ID`, "_"), y$X, y$`Model Sequence ID`)
y<-select(y, -X)
Ariba_data<-cbind(Ariba_data,y)

Ariba_data<-merge(Ariba_data, select(aro_index,`Model Sequence ID`, `AMR Gene Family`,`Drug Class`, `Resistance Mechanism` ) , by="Model Sequence ID")
```

Mapping quality?

### RPKM sum per AMR gene family

```{r}
library(knitr)
library(dplyr)
AMR_family_ARIBA<-Ariba_data%>%
  group_by(`AMR Gene Family`)%>%
  summarise('number of AMR genes per family(ARIBA)'=n(),
            'sum_RPKM_ariba'=sum(RPKM))%>%arrange(desc('sum_RPKM_ariba'))

kable(head(AMR_family_ARIBA))
```

# GROOT
Get groot report (uniqseq)

```{r}
library(readr)
Groot_data <- read_delim("groot_analysis/argannot-150bp-10000-reads-uniqseq-0report", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)
names(Groot_data)<-c("AMR.gene", "read.count", "gene.length", "coverage.cigar")
head(Groot_data)
```
Report: This will report gene, read count, gene length, coverage cigar"
Trasform GROOT notation and merge with CARD  DB
```{r}
library(stringr)
library(reshape2)
library(dplyr)
y<-colsplit(Groot_data$AMR.gene, "\\.", c("ARO Term", "ARO Acession","NCBI","init_final","Model Sequence ID","X"))

y$NCBI<-ifelse(str_detect(y$init_final, "-"),y$NCBI ,paste(y$NCBI,".",y$init_final) )
y$init_final<-ifelse(str_detect(y$init_final, "-"), y$init_final, y$`Model Sequence ID`)
y$`Model Sequence ID`<-ifelse(str_detect(y$`Model Sequence ID`, "-"), y$X, y$`Model Sequence ID`)
y<-select(y, -X)
Groot_data<-cbind(Groot_data,y)
Groot_data<-merge(Groot_data, select(aro_index,`Model Sequence ID`, `AMR Gene Family`,`Drug Class`, `Resistance Mechanism` ) , by="Model Sequence ID")
```

Calculate RPKM:

```{r}
Groot_data<-Groot_data%>%mutate(RPKM=read.count /(gene.length/1000 * totalNumReads/1000000))
```

```{r}
library(knitr)
n_distinct(Groot_data$`AMR Gene Family`)
AMR_family_Groot<-Groot_data%>%
  group_by(`AMR Gene Family`)%>%
  summarise('number of AMR genes per family (groot)'=n(),
            sum_RPKM_groot=sum(RPKM))%>%arrange(desc(sum_RPKM_groot))
kable(head(AMR_family_Groot))
```


# ARIBA vs RGI vs GROOT

```{r}
length(AMR_family_ARIBA$`AMR Gene Family`)
length(AMR_family_RGI$`AMR Gene Family`)
length(AMR_family_Groot$`AMR Gene Family`)
AMR_family<-merge(AMR_family_ARIBA, AMR_family_RGI[ ,1:3], by="AMR Gene Family", all=TRUE)
AMR_family<-merge(AMR_family, AMR_family_Groot, by="AMR Gene Family", all=TRUE)
```

```{r}
write_tsv(AMR_family, "AMR_family_ARIBAall_groot_uniqseq.tsv")
```


```{r}
venntransform<-function(x){
  x$RGI<-ifelse(is.na(x$sum_RPKM_RGI),FALSE,TRUE)
  x$Ariba<-ifelse(is.na(x$sum_RPKM_ariba),FALSE,TRUE)
  x$Groot<-ifelse(is.na(x$sum_RPKM_groot),FALSE,TRUE)
  x$All<-ifelse(x$RGI==TRUE & x$Ariba==TRUE & x$Groot==TRUE, TRUE, FALSE)
  return(x)}

library(ggvenn)
vennplot<-function(x){
y<-ggplot(x) + geom_venn(aes(A = RGI, B = Ariba, C = Groot),fill_color = c("chartreuse", "orange", "darkorchid1"), text_size=4, set_name_size = 2, stroke_size = 0.2, show_percentage = FALSE)+theme_void()+coord_fixed()+theme(title=element_text(size=5))
  return(y)}
```

```{r}
AMR_family_venn<-venntransform(AMR_family)
vennplot(AMR_family_venn)
```

```{r}
data_trans<-function(x){
x$RGI<-ifelse(is.na(x$sum_RPKM_RGI),0,1)
x$Ariba<-ifelse(is.na(x$sum_RPKM_ariba),0,1)
x$Groot<-ifelse(is.na(x$sum_RPKM_groot),0,1)
x$npipelines<-apply(select(x, RGI, Ariba,Groot),1, sum)
x$npipelines<-as.factor(x$npipelines)
x$pipelines<-ifelse(x$npipelines==3, "All", 
                                ifelse(x$npipelines==2 & x$RGI==0, "Groot+Ariba",
                                       ifelse(x$npipelines==2 & x$Ariba==0, "RGI+Groot",
                                              ifelse(x$npipelines==2 & x$Groot==0, "RGI+Ariba",
                                                     ifelse(x$RGI==1, "RGI only",
                                                            ifelse(x$Ariba==1, "Ariba only", "Groot only"))))))
return(x)}
```

```{r}
AMR_family<-data_trans(AMR_family)

AMR_family[AMR_family$pipelines=="RGI only" ,1]
AMR_family[AMR_family$pipelines=="Groot only" ,1]
AMR_family[AMR_family$pipelines=="RGI+Groot" ,1]
AMR_family[AMR_family$pipelines=="Groot+Ariba" ,1]
```


```{r}
AMR_family$pipelines<-factor(AMR_family$pipelines, ordered = TRUE, levels = c("RGI only", "Ariba only", "Groot only", "Groot+Ariba", "RGI+Groot", "RGI+Ariba", "All"))

p_RGI<-ggplot(subset(AMR_family, AMR_family$RGI==1), aes(pipelines, sum_RPKM_RGI))+geom_boxplot(aes(colour=npipelines), outlier.shape = NA, na.rm = TRUE)+geom_jitter(width = 0.2, aes(colour=npipelines))+theme_classic()+theme(legend.position = "none")+ggtitle("RGI")

p_Ariba<-ggplot(subset(AMR_family, AMR_family$Ariba==1), aes(pipelines, sum_RPKM_ariba))+geom_boxplot(aes(colour=npipelines), outlier.shape = NA, na.rm = TRUE)+geom_jitter(width = 0.2, aes(colour=npipelines))+theme_classic()+theme(legend.position = "none")+ggtitle("Ariba")

p_Groot<-ggplot(subset(AMR_family, AMR_family$Groot==1), aes(pipelines, sum_RPKM_groot))+geom_boxplot(aes(colour=npipelines), outlier.shape = NA, na.rm = TRUE)+geom_jitter(width = 0.2, aes(colour=npipelines))+theme_classic()+theme(legend.position = "none")+ggtitle("Groot")
p_RGI
p_Ariba
p_Groot
```



