% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{article}
\usepackage{amsmath,amssymb}
\usepackage{lmodern}
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\hypersetup{
  pdftitle={RPKM},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}
\urlstyle{same} % disable monospaced font for URLs
\usepackage[margin=1in]{geometry}
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.77,0.63,0.00}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\usepackage{longtable,booktabs,array}
\usepackage{calc} % for calculating minipage widths
% Correct order of tables after \paragraph or \subparagraph
\usepackage{etoolbox}
\makeatletter
\patchcmd\longtable{\par}{\if@noskipsec\mbox{}\fi\par}{}{}
\makeatother
% Allow footnotes in longtable head/foot
\IfFileExists{footnotehyper.sty}{\usepackage{footnotehyper}}{\usepackage{footnote}}
\makesavenoteenv{longtable}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{-\maxdimen} % remove section numbering
\ifLuaTeX
  \usepackage{selnolig}  % disable illegal ligatures
\fi

\title{RPKM}
\author{}
\date{\vspace{-2.5em}}

\begin{document}
\maketitle

{
\setcounter{tocdepth}{2}
\tableofcontents
}
\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(stringr)}
\FunctionTok{library}\NormalTok{(reshape2)}
\FunctionTok{library}\NormalTok{(readr)}
\FunctionTok{library}\NormalTok{(dplyr)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
## Attaching package: 'dplyr'
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:stats':
## 
##     filter, lag
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:base':
## 
##     intersect, setdiff, setequal, union
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(patchwork)}
\FunctionTok{library}\NormalTok{(ggplot2)}
\FunctionTok{library}\NormalTok{(knitr)}
\end{Highlighting}
\end{Shaded}

Import CARD data aro index:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{aro\_index }\OtherTok{\textless{}{-}} \FunctionTok{read\_delim}\NormalTok{(}\StringTok{"PRJNA307231/aro\_index.tsv"}\NormalTok{, }
    \StringTok{"}\SpecialCharTok{\textbackslash{}t}\StringTok{"}\NormalTok{, }\AttributeTok{escape\_double =} \ConstantTok{FALSE}\NormalTok{, }\AttributeTok{trim\_ws =} \ConstantTok{TRUE}\NormalTok{)}

\FunctionTok{dim}\NormalTok{(aro\_index)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 3326   11
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{length}\NormalTok{(}\FunctionTok{unique}\NormalTok{(aro\_index}\SpecialCharTok{$}\StringTok{\textasciigrave{}}\AttributeTok{ARO Accession}\StringTok{\textasciigrave{}}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 3323
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{y}\OtherTok{\textless{}{-}}\FunctionTok{colsplit}\NormalTok{(aro\_index}\SpecialCharTok{$}\StringTok{\textasciigrave{}}\AttributeTok{ARO Accession}\StringTok{\textasciigrave{}}\NormalTok{, }\StringTok{":"}\NormalTok{, }\FunctionTok{c}\NormalTok{(}\StringTok{"X"}\NormalTok{, }\StringTok{"ARO Number"}\NormalTok{))}
\NormalTok{aro\_index}\OtherTok{\textless{}{-}}\FunctionTok{cbind}\NormalTok{(aro\_index, }\FunctionTok{select}\NormalTok{(y, }\StringTok{\textasciigrave{}}\AttributeTok{ARO Number}\StringTok{\textasciigrave{}}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\hypertarget{calculate-rpkm}{%
\section{Calculate RPKM}\label{calculate-rpkm}}

\hypertarget{rgi}{%
\subsection{RGI}\label{rgi}}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(readr)}
\NormalTok{RGI\_data }\OtherTok{\textless{}{-}} \FunctionTok{read\_delim}\NormalTok{(}\StringTok{"data\_testing/RGI\_analysis/argannot{-}150bp{-}10000{-}reads.fastq\_RGI\_output.allele\_mapping\_data.txt"}\NormalTok{, }\StringTok{"}\SpecialCharTok{\textbackslash{}t}\StringTok{"}\NormalTok{, }\AttributeTok{escape\_double =} \ConstantTok{FALSE}\NormalTok{, }\AttributeTok{trim\_ws =} \ConstantTok{TRUE}\NormalTok{)}
\FunctionTok{kable}\NormalTok{(}\FunctionTok{head}\NormalTok{(RGI\_data))}
\end{Highlighting}
\end{Shaded}

\begin{longtable}[]{@{}
  >{\raggedright\arraybackslash}p{(\columnwidth - 38\tabcolsep) * \real{0.10}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 38\tabcolsep) * \real{0.02}}
  >{\raggedleft\arraybackslash}p{(\columnwidth - 38\tabcolsep) * \real{0.02}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 38\tabcolsep) * \real{0.04}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 38\tabcolsep) * \real{0.02}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 38\tabcolsep) * \real{0.04}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 38\tabcolsep) * \real{0.07}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 38\tabcolsep) * \real{0.08}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 38\tabcolsep) * \real{0.07}}
  >{\raggedleft\arraybackslash}p{(\columnwidth - 38\tabcolsep) * \real{0.04}}
  >{\raggedleft\arraybackslash}p{(\columnwidth - 38\tabcolsep) * \real{0.06}}
  >{\raggedleft\arraybackslash}p{(\columnwidth - 38\tabcolsep) * \real{0.03}}
  >{\raggedleft\arraybackslash}p{(\columnwidth - 38\tabcolsep) * \real{0.03}}
  >{\raggedleft\arraybackslash}p{(\columnwidth - 38\tabcolsep) * \real{0.03}}
  >{\raggedleft\arraybackslash}p{(\columnwidth - 38\tabcolsep) * \real{0.06}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 38\tabcolsep) * \real{0.03}}
  >{\raggedleft\arraybackslash}p{(\columnwidth - 38\tabcolsep) * \real{0.03}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 38\tabcolsep) * \real{0.07}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 38\tabcolsep) * \real{0.12}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 38\tabcolsep) * \real{0.05}}@{}}
\toprule
Reference Sequence & ARO Term & ARO Accession & Reference Model Type &
Reference DB & Reference Allele Source & Resistomes \& Variants:
Observed in Genome(s) & Resistomes \& Variants: Observed in Plasmid(s) &
Resistomes \& Variants: Observed Pathogen(s) & Completely Mapped Reads &
Mapped Reads with Flanking Sequence & All Mapped Reads & Percent
Coverage & Length Coverage (bp) & Average MAPQ (Completely Mapped Reads)
& Mate Pair Linkage & Reference Length & AMR Gene Family & Drug Class &
Resistance Mechanism \\
\midrule
\endhead
ARO:3001109\textbar ID:4\textbar Name:SHV-52\textbar NCBI:HQ845196 &
SHV-52 & 3001109 & protein homolog model & CARD & CARD curation & no
data & no data & Klebsiella pneumoniae & 5 & 0 & 5 & 60.98 & 525 & 1 & *
& 861 & SHV beta-lactamase & carbapenem; cephalosporin; penam &
antibiotic inactivation \\
ARO:3001989\textbar ID:7\textbar Name:CTX-M-130\textbar NCBI:JX017365 &
CTX-M-130 & 3001989 & protein homolog model & CARD & CARD curation & no
data & no data & Escherichia coli & 2 & 0 & 2 & 34.25 & 300 & 1 & * &
876 & CTX-M beta-lactamase & cephalosporin & antibiotic inactivation \\
ARO:3002356\textbar ID:8\textbar Name:NDM-6\textbar NCBI:JN967644 &
NDM-6 & 3002356 & protein homolog model & CARD & CARD curation & no data
& no data & Escherichia coli & 4 & 0 & 4 & 57.69 & 469 & 1 & * & 813 &
NDM beta-lactamase & carbapenem; cephalosporin; cephamycin; penam &
antibiotic inactivation \\
ARO:3001855\textbar ID:9\textbar Name:ACT-35\textbar NCBI:LC004922 &
ACT-35 & 3001855 & protein homolog model & CARD & CARD curation & no
data & no data & Enterobacter cloacae & 2 & 0 & 2 & 26.18 & 300 & 1 & *
& 1146 & ACT beta-lactamase & carbapenem; cephalosporin; cephamycin;
penam & antibiotic inactivation \\
ARO:3002244\textbar ID:10\textbar Name:CARB-5\textbar NCBI:AF135373 &
CARB-5 & 3002244 & protein homolog model & CARD & CARD curation & no
data & no data & Acinetobacter calcoaceticus & 4 & 0 & 4 & 58.86 & 528 &
1 & * & 897 & CARB beta-lactamase & penam & antibiotic inactivation \\
ARO:3000600\textbar ID:11\textbar Name:Erm(34)\textbar NCBI:AY234334 &
Erm(34) & 3000600 & protein homolog model & CARD & CARD curation & no
data & no data & Bacillus clausii & 6 & 0 & 6 & 58.75 & 497 & 44 & * &
846 & Erm 23S ribosomal RNA methyltransferase & macrolide antibiotic;
lincosamide antibiotic; streptogramin antibiotic & antibiotic target
alteration \\
\bottomrule
\end{longtable}

Obtain AMR gene family from CARD database:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{RGI\_data}\OtherTok{\textless{}{-}}\FunctionTok{merge}\NormalTok{(}
\FunctionTok{select}\NormalTok{(RGI\_data, }\SpecialCharTok{{-}}\StringTok{\textasciigrave{}}\AttributeTok{AMR Gene Family}\StringTok{\textasciigrave{}}\NormalTok{, }\SpecialCharTok{{-}}\StringTok{\textasciigrave{}}\AttributeTok{Drug Class}\StringTok{\textasciigrave{}}\NormalTok{, }\SpecialCharTok{{-}}\StringTok{\textasciigrave{}}\AttributeTok{Resistance Mechanism}\StringTok{\textasciigrave{}}\NormalTok{),}
\FunctionTok{select}\NormalTok{(aro\_index,}\StringTok{\textasciigrave{}}\AttributeTok{Model Sequence ID}\StringTok{\textasciigrave{}}\NormalTok{, }\StringTok{\textasciigrave{}}\AttributeTok{AMR Gene Family}\StringTok{\textasciigrave{}}\NormalTok{,}\StringTok{\textasciigrave{}}\AttributeTok{Drug Class}\StringTok{\textasciigrave{}}\NormalTok{, }\StringTok{\textasciigrave{}}\AttributeTok{Resistance Mechanism}\StringTok{\textasciigrave{}}\NormalTok{,}\StringTok{\textasciigrave{}}\AttributeTok{Protein Accession}\StringTok{\textasciigrave{}}\NormalTok{,}\StringTok{\textasciigrave{}}\AttributeTok{ARO Number}\StringTok{\textasciigrave{}}\NormalTok{, }\StringTok{\textasciigrave{}}\AttributeTok{ARO Name}\StringTok{\textasciigrave{}}\NormalTok{ ), }\AttributeTok{by.x =} \StringTok{"ARO Accession"}\NormalTok{, }\AttributeTok{by.y=}\StringTok{"ARO Number"}\NormalTok{, }\AttributeTok{all.x =} \ConstantTok{TRUE}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{table}\NormalTok{(}\FunctionTok{is.na}\NormalTok{(RGI\_data}\SpecialCharTok{$}\StringTok{\textasciigrave{}}\AttributeTok{AMR Gene Family}\StringTok{\textasciigrave{}}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
## FALSE 
##  2019
\end{verbatim}

Calculate RPKM:

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(dplyr)}
\NormalTok{totalNumReads}\OtherTok{\textless{}{-}} \DecValTok{10000}
\NormalTok{RGI\_data}\OtherTok{\textless{}{-}}\NormalTok{RGI\_data}\SpecialCharTok{\%\textgreater{}\%}\FunctionTok{mutate}\NormalTok{(}\AttributeTok{RPKM=}\StringTok{\textasciigrave{}}\AttributeTok{All Mapped Reads}\StringTok{\textasciigrave{}} \SpecialCharTok{/}\NormalTok{(}\StringTok{\textasciigrave{}}\AttributeTok{Reference Length}\StringTok{\textasciigrave{}}\SpecialCharTok{/}\DecValTok{1000} \SpecialCharTok{*}\NormalTok{ totalNumReads}\SpecialCharTok{/}\DecValTok{1000000}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\hypertarget{average-mapq-distribution}{%
\subsubsection{Average MAPQ
distribution}\label{average-mapq-distribution}}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(ggplot2)}
\FunctionTok{ggplot}\NormalTok{(RGI\_data, }\FunctionTok{aes}\NormalTok{(}\StringTok{\textasciigrave{}}\AttributeTok{Average MAPQ (Completely Mapped Reads)}\StringTok{\textasciigrave{}}\NormalTok{)) }\SpecialCharTok{+}
  \FunctionTok{geom\_histogram}\NormalTok{(}\AttributeTok{bins=}\DecValTok{50}\NormalTok{, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{y =} \FunctionTok{stat}\NormalTok{(density)))}\SpecialCharTok{+}\FunctionTok{geom\_density}\NormalTok{()}\SpecialCharTok{+}\FunctionTok{theme\_bw}\NormalTok{()}
\end{Highlighting}
\end{Shaded}

\includegraphics{test_data_rpkm_files/figure-latex/unnamed-chunk-7-1.pdf}

\hypertarget{apply-average-mapq-filter}{%
\subsubsection{Apply Average MAPQ
filter:}\label{apply-average-mapq-filter}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{RGI\_data}\OtherTok{\textless{}{-}}\NormalTok{RGI\_data}\SpecialCharTok{\%\textgreater{}\%}\FunctionTok{filter}\NormalTok{(}\StringTok{\textasciigrave{}}\AttributeTok{Average MAPQ (Completely Mapped Reads)}\StringTok{\textasciigrave{}}\SpecialCharTok{\textgreater{}}\DecValTok{10}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\hypertarget{rpkm-sum-per-amr-gene-family}{%
\subsubsection{RPKM sum per AMR gene
family}\label{rpkm-sum-per-amr-gene-family}}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(knitr)}
\FunctionTok{n\_distinct}\NormalTok{(RGI\_data}\SpecialCharTok{$}\StringTok{\textasciigrave{}}\AttributeTok{AMR Gene Family}\StringTok{\textasciigrave{}}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 125
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{AMR\_family\_RGI}\OtherTok{\textless{}{-}}\NormalTok{RGI\_data}\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{group\_by}\NormalTok{(}\StringTok{\textasciigrave{}}\AttributeTok{AMR Gene Family}\StringTok{\textasciigrave{}}\NormalTok{)}\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{summarise}\NormalTok{(}\StringTok{\textquotesingle{}number of AMR genes per family (RGI)\textquotesingle{}}\OtherTok{=}\FunctionTok{n}\NormalTok{(),}
            \AttributeTok{sum\_RPKM\_RGI=}\FunctionTok{sum}\NormalTok{(RPKM),}
            \AttributeTok{avg\_MAPQ=}\FunctionTok{mean}\NormalTok{(}\StringTok{\textasciigrave{}}\AttributeTok{Average MAPQ (Completely Mapped Reads)}\StringTok{\textasciigrave{}}\NormalTok{),}
            \AttributeTok{q1\_MAPQ=}\FunctionTok{quantile}\NormalTok{(}\StringTok{\textasciigrave{}}\AttributeTok{Average MAPQ (Completely Mapped Reads)}\StringTok{\textasciigrave{}}\NormalTok{, }\FloatTok{0.25}\NormalTok{),}
            \AttributeTok{p3\_MAPQ=}\FunctionTok{quantile}\NormalTok{(}\StringTok{\textasciigrave{}}\AttributeTok{Average MAPQ (Completely Mapped Reads)}\StringTok{\textasciigrave{}}\NormalTok{,}\FloatTok{0.75}\NormalTok{),)}\SpecialCharTok{\%\textgreater{}\%}\FunctionTok{arrange}\NormalTok{(}\FunctionTok{desc}\NormalTok{(sum\_RPKM\_RGI))}
\FunctionTok{kable}\NormalTok{(}\FunctionTok{head}\NormalTok{(AMR\_family\_RGI))}
\end{Highlighting}
\end{Shaded}

\begin{longtable}[]{@{}
  >{\raggedright\arraybackslash}p{(\columnwidth - 10\tabcolsep) * \real{0.44}}
  >{\raggedleft\arraybackslash}p{(\columnwidth - 10\tabcolsep) * \real{0.28}}
  >{\raggedleft\arraybackslash}p{(\columnwidth - 10\tabcolsep) * \real{0.10}}
  >{\raggedleft\arraybackslash}p{(\columnwidth - 10\tabcolsep) * \real{0.07}}
  >{\raggedleft\arraybackslash}p{(\columnwidth - 10\tabcolsep) * \real{0.06}}
  >{\raggedleft\arraybackslash}p{(\columnwidth - 10\tabcolsep) * \real{0.06}}@{}}
\toprule
AMR Gene Family & number of AMR genes per family (RGI) & sum\_RPKM\_RGI
& avg\_MAPQ & q1\_MAPQ & p3\_MAPQ \\
\midrule
\endhead
trimethoprim resistant dihydrofolate reductase dfr & 35 & 22523.60 &
35.09714 & 31.00 & 44 \\
major facilitator superfamily (MFS) antibiotic efflux pump & 36 &
22513.83 & 41.25917 & 44.00 & 44 \\
OXA beta-lactamase & 41 & 21847.96 & 25.99024 & 11.00 & 35 \\
Erm 23S ribosomal RNA methyltransferase & 29 & 19076.80 & 42.06690 &
41.71 & 44 \\
AAC(6') & 26 & 16687.01 & 36.13154 & 32.68 & 44 \\
chloramphenicol acetyltransferase (CAT) & 21 & 15610.47 & 36.57667 &
31.25 & 44 \\
\bottomrule
\end{longtable}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{str}\NormalTok{(AMR\_family\_RGI}\SpecialCharTok{$}\StringTok{\textasciigrave{}}\AttributeTok{sum RPKM (RGI)}\StringTok{\textasciigrave{}}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning: Unknown or uninitialised column: `sum RPKM (RGI)`.
\end{verbatim}

\begin{verbatim}
##  NULL
\end{verbatim}

\hypertarget{ariba}{%
\section{ARIBA}\label{ariba}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{Ariba\_data }\OtherTok{\textless{}{-}} \FunctionTok{read\_delim}\NormalTok{(}\StringTok{"data\_testing/ariba\_analysis/argannot{-}150bp{-}10000{-}reads\_ariba\_results/report.tsv"}\NormalTok{, }
    \StringTok{"}\SpecialCharTok{\textbackslash{}t}\StringTok{"}\NormalTok{, }\AttributeTok{escape\_double =} \ConstantTok{FALSE}\NormalTok{, }\AttributeTok{trim\_ws =} \ConstantTok{TRUE}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Remove contig and variant data, select only coding sequences

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{Ariba\_data}\OtherTok{\textless{}{-}}\NormalTok{Ariba\_data }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{(gene}\SpecialCharTok{==}\DecValTok{1}\NormalTok{)}\SpecialCharTok{\%\textgreater{}\%}\FunctionTok{select}\NormalTok{(}\FunctionTok{c}\NormalTok{(}\DecValTok{1}\SpecialCharTok{:}\DecValTok{8}\NormalTok{,}\DecValTok{31}\NormalTok{))}\SpecialCharTok{\%\textgreater{}\%}\FunctionTok{unique}\NormalTok{()}
\end{Highlighting}
\end{Shaded}

Calculate RPKM:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{Ariba\_data}\OtherTok{\textless{}{-}}\NormalTok{Ariba\_data}\SpecialCharTok{\%\textgreater{}\%}\FunctionTok{mutate}\NormalTok{(}\AttributeTok{RPKM=}\NormalTok{reads }\SpecialCharTok{/}\NormalTok{(ref\_len}\SpecialCharTok{/}\DecValTok{1000} \SpecialCharTok{*}\NormalTok{ totalNumReads}\SpecialCharTok{/}\DecValTok{1000000}\NormalTok{))}
\FunctionTok{kable}\NormalTok{(}\FunctionTok{head}\NormalTok{(Ariba\_data))}
\end{Highlighting}
\end{Shaded}

\begin{longtable}[]{@{}
  >{\raggedright\arraybackslash}p{(\columnwidth - 18\tabcolsep) * \real{0.16}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 18\tabcolsep) * \real{0.16}}
  >{\raggedleft\arraybackslash}p{(\columnwidth - 18\tabcolsep) * \real{0.02}}
  >{\raggedleft\arraybackslash}p{(\columnwidth - 18\tabcolsep) * \real{0.03}}
  >{\raggedleft\arraybackslash}p{(\columnwidth - 18\tabcolsep) * \real{0.02}}
  >{\raggedleft\arraybackslash}p{(\columnwidth - 18\tabcolsep) * \real{0.02}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 18\tabcolsep) * \real{0.04}}
  >{\raggedleft\arraybackslash}p{(\columnwidth - 18\tabcolsep) * \real{0.03}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 18\tabcolsep) * \real{0.49}}
  >{\raggedleft\arraybackslash}p{(\columnwidth - 18\tabcolsep) * \real{0.03}}@{}}
\toprule
\#ariba\_ref\_name & ref\_name & gene & var\_only & flag & reads &
cluster & ref\_len & free\_text & RPKM \\
\midrule
\endhead
AAC\_2\_\_\_Ic.3002525.AL123456.3.314308\_314854.4719 &
AAC\_2\_\_\_Ic.3002525.AL123456.3.314308\_314854.4719 & 1 & 0 & 16 & 12
& AAC\_2\_\_\_Ic & 546 & b``AAC(2')-Ic is a chromosomal-encoded
aminoglycoside acetyltransferase in M. tuberculosis and Mycobacterium
tuberculosis variant bovis'';AAC(2')-Ic & 2197.802 \\
AAC\_2\_\_\_Ie.3002527.AL583926.1.40191\_40740.5510 &
AAC\_2\_\_\_Ie.3002527.AL583926.1.40191\_40740.5510 & 1 & 0 & 16 & 12 &
AAC\_2\_\_\_Ie & 549 & b``AAC(2')-Ie is a chromosomal-encoded
aminoglycoside acetyltransferase in M. leprae'';AAC(2')-Ie & 2185.792 \\
AAC\_3\_\_IIb.3002534.M97172.655\_1465.10 &
AAC\_3\_\_IIb.3002534.M97172.655\_1465.10 & 1 & 0 & 16 & 20 &
AAC\_3\_\_IIb & 810 & AAC(3)-IIb;b'AAC(3)-IIb is an aminoglycoside
acetyltransferase in E. coli, A. faecalis and S. marcescens' &
2469.136 \\
AAC\_3\_\_IIe.3004621.EU022315.1.0\_861.5340 &
AAC\_3\_\_IIe.3004621.EU022315.1.0\_861.5340 & 1 & 0 & 16 & 50 &
AAC\_3\_\_II- & 861 & b'AAC(3)-IIe is a plasmid-encoded aminoglycoside
acetyltransferase in E. coli';AAC(3)-IIe & 5807.201 \\
AAC\_3\_\_IV.3002539.DQ241380.1.0\_777.3715 &
AAC\_3\_\_IV.3002539.DQ241380.1.0\_777.3715 & 1 & 0 & 16 & 16 &
AAC\_3\_\_IV & 777 & b'AAC(3)-IV is a plasmid-encoded aminoglycoside
acetyltransferase in E. coli, C. jejuni and P. stutzeri';AAC(3)-IV &
2059.202 \\
AAC\_3\_\_VIIIa.3002542.M55426.465\_1326.12 &
AAC\_3\_\_VIIIa.3002542.M55426.465\_1326.12 & 1 & 0 & 528 & 14 &
AAC\_3\_\_VIIIa & 861 & b'AAC(3)-VIIIa is a chromosomal-encoded
aminoglycoside acetyltransferase in Streptomyces fradiae';AAC(3)-VIIIa &
1626.016 \\
\bottomrule
\end{longtable}

\hypertarget{obtain-amr-gene-family}{%
\subsubsection{Obtain AMR gene family}\label{obtain-amr-gene-family}}

Trasform ARIBA notation to merge with CARD DB

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(stringr)}
\FunctionTok{library}\NormalTok{(reshape2)}
\NormalTok{y}\OtherTok{\textless{}{-}}\FunctionTok{colsplit}\NormalTok{(Ariba\_data}\SpecialCharTok{$}\NormalTok{ref\_name, }\StringTok{"}\SpecialCharTok{\textbackslash{}\textbackslash{}}\StringTok{."}\NormalTok{, }\FunctionTok{c}\NormalTok{(}\StringTok{"ARO Term"}\NormalTok{, }\StringTok{"ARO Accession"}\NormalTok{,}\StringTok{"NCBI"}\NormalTok{,}\StringTok{"init\_final"}\NormalTok{,}\StringTok{"Model Sequence ID"}\NormalTok{,}\StringTok{"X"}\NormalTok{))}
\NormalTok{y}\SpecialCharTok{$}\NormalTok{NCBI}\OtherTok{\textless{}{-}}\FunctionTok{ifelse}\NormalTok{(}\FunctionTok{str\_detect}\NormalTok{(y}\SpecialCharTok{$}\NormalTok{init\_final, }\StringTok{"\_"}\NormalTok{),y}\SpecialCharTok{$}\NormalTok{NCBI ,}\FunctionTok{paste}\NormalTok{(y}\SpecialCharTok{$}\NormalTok{NCBI,}\StringTok{"."}\NormalTok{,y}\SpecialCharTok{$}\NormalTok{init\_final) )}
\NormalTok{y}\SpecialCharTok{$}\NormalTok{init\_final}\OtherTok{\textless{}{-}}\FunctionTok{ifelse}\NormalTok{(}\FunctionTok{str\_detect}\NormalTok{(y}\SpecialCharTok{$}\NormalTok{init\_final, }\StringTok{"\_"}\NormalTok{), y}\SpecialCharTok{$}\NormalTok{init\_final, y}\SpecialCharTok{$}\StringTok{\textasciigrave{}}\AttributeTok{Model Sequence ID}\StringTok{\textasciigrave{}}\NormalTok{)}
\NormalTok{y}\SpecialCharTok{$}\StringTok{\textasciigrave{}}\AttributeTok{Model Sequence ID}\StringTok{\textasciigrave{}}\OtherTok{\textless{}{-}}\FunctionTok{ifelse}\NormalTok{(}\FunctionTok{str\_detect}\NormalTok{(y}\SpecialCharTok{$}\StringTok{\textasciigrave{}}\AttributeTok{Model Sequence ID}\StringTok{\textasciigrave{}}\NormalTok{, }\StringTok{"\_"}\NormalTok{), y}\SpecialCharTok{$}\NormalTok{X, y}\SpecialCharTok{$}\StringTok{\textasciigrave{}}\AttributeTok{Model Sequence ID}\StringTok{\textasciigrave{}}\NormalTok{)}
\NormalTok{y}\SpecialCharTok{$}\StringTok{\textasciigrave{}}\AttributeTok{ARO Accession}\StringTok{\textasciigrave{}}\OtherTok{\textless{}{-}}\FunctionTok{ifelse}\NormalTok{(y}\SpecialCharTok{$}\StringTok{\textasciigrave{}}\AttributeTok{ARO Accession}\StringTok{\textasciigrave{}}\SpecialCharTok{\textless{}}\DecValTok{3000000}\NormalTok{, y}\SpecialCharTok{$}\NormalTok{NCBI, y}\SpecialCharTok{$}\StringTok{\textasciigrave{}}\AttributeTok{ARO Accession}\StringTok{\textasciigrave{}}\NormalTok{)}
\NormalTok{y}\SpecialCharTok{$}\StringTok{\textasciigrave{}}\AttributeTok{ARO Accession}\StringTok{\textasciigrave{}}\OtherTok{\textless{}{-}}\FunctionTok{str\_split\_fixed}\NormalTok{(y}\SpecialCharTok{$}\StringTok{\textasciigrave{}}\AttributeTok{ARO Accession}\StringTok{\textasciigrave{}}\NormalTok{, }\StringTok{"}\SpecialCharTok{\textbackslash{}\textbackslash{}}\StringTok{."}\NormalTok{, }\DecValTok{2}\NormalTok{)[ ,}\DecValTok{1}\NormalTok{]}
\NormalTok{y}\OtherTok{\textless{}{-}}\FunctionTok{select}\NormalTok{(y, }\SpecialCharTok{{-}}\NormalTok{X)}
\NormalTok{Ariba\_data}\OtherTok{\textless{}{-}}\FunctionTok{cbind}\NormalTok{(Ariba\_data,y)}
\NormalTok{Ariba\_data}\SpecialCharTok{$}\StringTok{\textasciigrave{}}\AttributeTok{ARO Accession}\StringTok{\textasciigrave{}}\OtherTok{\textless{}{-}}\FunctionTok{as.numeric}\NormalTok{(Ariba\_data}\SpecialCharTok{$}\StringTok{\textasciigrave{}}\AttributeTok{ARO Accession}\StringTok{\textasciigrave{}}\NormalTok{)}

\NormalTok{Ariba\_data}\OtherTok{\textless{}{-}}\FunctionTok{merge}\NormalTok{(Ariba\_data, }\FunctionTok{select}\NormalTok{(aro\_index,}\StringTok{\textasciigrave{}}\AttributeTok{Model Sequence ID}\StringTok{\textasciigrave{}}\NormalTok{, }\StringTok{\textasciigrave{}}\AttributeTok{AMR Gene Family}\StringTok{\textasciigrave{}}\NormalTok{,}\StringTok{\textasciigrave{}}\AttributeTok{Drug Class}\StringTok{\textasciigrave{}}\NormalTok{, }\StringTok{\textasciigrave{}}\AttributeTok{Resistance Mechanism}\StringTok{\textasciigrave{}}\NormalTok{,}\StringTok{\textasciigrave{}}\AttributeTok{Protein Accession}\StringTok{\textasciigrave{}}\NormalTok{,  }\StringTok{\textasciigrave{}}\AttributeTok{ARO Number}\StringTok{\textasciigrave{}}\NormalTok{, }\StringTok{\textasciigrave{}}\AttributeTok{ARO Name}\StringTok{\textasciigrave{}}\NormalTok{) , }\AttributeTok{by.x =} \StringTok{"ARO Accession"}\NormalTok{, }\AttributeTok{by.y=}\StringTok{"ARO Number"}\NormalTok{, }\AttributeTok{all.x =} \ConstantTok{TRUE}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\hypertarget{rpkm-sum-per-amr-gene-family-1}{%
\subsubsection{RPKM sum per AMR gene
family}\label{rpkm-sum-per-amr-gene-family-1}}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(knitr)}
\FunctionTok{library}\NormalTok{(dplyr)}
\NormalTok{AMR\_family\_ARIBA}\OtherTok{\textless{}{-}}\NormalTok{Ariba\_data}\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{group\_by}\NormalTok{(}\StringTok{\textasciigrave{}}\AttributeTok{AMR Gene Family}\StringTok{\textasciigrave{}}\NormalTok{)}\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{summarise}\NormalTok{(}\StringTok{\textquotesingle{}number of AMR genes per family(ARIBA)\textquotesingle{}}\OtherTok{=}\FunctionTok{n}\NormalTok{(),}
            \StringTok{\textquotesingle{}sum\_RPKM\_ariba\textquotesingle{}}\OtherTok{=}\FunctionTok{sum}\NormalTok{(RPKM))}\SpecialCharTok{\%\textgreater{}\%}\FunctionTok{arrange}\NormalTok{(}\FunctionTok{desc}\NormalTok{(}\StringTok{\textquotesingle{}sum\_RPKM\_ariba\textquotesingle{}}\NormalTok{))}

\FunctionTok{kable}\NormalTok{(}\FunctionTok{head}\NormalTok{(AMR\_family\_ARIBA))}
\end{Highlighting}
\end{Shaded}

\begin{longtable}[]{@{}lrr@{}}
\toprule
AMR Gene Family & number of AMR genes per family(ARIBA) &
sum\_RPKM\_ariba \\
\midrule
\endhead
16S rRNA methyltransferase (A1408) & 1 & 1818.182 \\
16S rRNA methyltransferase (G1405) & 2 & 2797.478 \\
AAC(2') & 2 & 4383.595 \\
AAC(3) & 5 & 14768.573 \\
AAC(6') & 6 & 18053.867 \\
AAC(6');APH(2'\,') & 1 & 1666.667 \\
\bottomrule
\end{longtable}

\hypertarget{groot}{%
\section{GROOT}\label{groot}}

Get groot report (uniqseq)

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(readr)}
\NormalTok{Groot\_data }\OtherTok{\textless{}{-}} \FunctionTok{read\_delim}\NormalTok{(}\StringTok{"data\_testing/groot\_analysis/argannot{-}150bp{-}10000{-}reads{-}uniqseq{-}0report"}\NormalTok{, }
    \StringTok{"}\SpecialCharTok{\textbackslash{}t}\StringTok{"}\NormalTok{, }\AttributeTok{escape\_double =} \ConstantTok{FALSE}\NormalTok{, }\AttributeTok{col\_names =} \ConstantTok{FALSE}\NormalTok{, }
    \AttributeTok{trim\_ws =} \ConstantTok{TRUE}\NormalTok{)}
\FunctionTok{names}\NormalTok{(Groot\_data)}\OtherTok{\textless{}{-}}\FunctionTok{c}\NormalTok{(}\StringTok{"AMR.gene"}\NormalTok{, }\StringTok{"read.count"}\NormalTok{, }\StringTok{"gene.length"}\NormalTok{, }\StringTok{"coverage.cigar"}\NormalTok{)}
\FunctionTok{kable}\NormalTok{(}\FunctionTok{head}\NormalTok{(Groot\_data))}
\end{Highlighting}
\end{Shaded}

\begin{longtable}[]{@{}
  >{\raggedright\arraybackslash}p{(\columnwidth - 6\tabcolsep) * \real{0.58}}
  >{\raggedleft\arraybackslash}p{(\columnwidth - 6\tabcolsep) * \real{0.10}}
  >{\raggedleft\arraybackslash}p{(\columnwidth - 6\tabcolsep) * \real{0.10}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 6\tabcolsep) * \real{0.22}}@{}}
\toprule
AMR.gene & read.count & gene.length & coverage.cigar \\
\midrule
\endhead
Bacillus\_clausii\_chloramphenicol.3004452.AY238971.1.1054-1741.4424 & 4
& 687 & 198D338M151D \\
OXA-474.3003631.KR182167.1.0-810.5731 & 1 & 810 & 538D151M121D \\
TEM-92.3000959.AF143804.0-861.1961 & 4 & 861 & 282D151M142D254M32D \\
OXA-67.3001437.DQ491200.0-825.1936 & 2 & 825 & 7D151M118D151M398D \\
TEM-216.3001393.KF944358.0-861.1494 & 3 & 861 & 447D151M43D201M19D \\
rmtD.3002667.EU269034.2.1479-2223.4675 & 3 & 744 &
45D151M87D151M61D151M98D \\
\bottomrule
\end{longtable}

Trasform GROOT notation and merge with CARD DB

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(stringr)}
\FunctionTok{library}\NormalTok{(reshape2)}
\FunctionTok{library}\NormalTok{(dplyr)}
\NormalTok{y}\OtherTok{\textless{}{-}}\FunctionTok{colsplit}\NormalTok{(Groot\_data}\SpecialCharTok{$}\NormalTok{AMR.gene, }\StringTok{"}\SpecialCharTok{\textbackslash{}\textbackslash{}}\StringTok{."}\NormalTok{, }\FunctionTok{c}\NormalTok{(}\StringTok{"ARO Term"}\NormalTok{, }\StringTok{"ARO Accession"}\NormalTok{,}\StringTok{"NCBI"}\NormalTok{,}\StringTok{"init\_final"}\NormalTok{,}\StringTok{"Model Sequence ID"}\NormalTok{,}\StringTok{"X"}\NormalTok{))}
\NormalTok{y}\SpecialCharTok{$}\NormalTok{NCBI}\OtherTok{\textless{}{-}}\FunctionTok{ifelse}\NormalTok{(}\FunctionTok{str\_detect}\NormalTok{(y}\SpecialCharTok{$}\NormalTok{init\_final, }\StringTok{"{-}"}\NormalTok{),y}\SpecialCharTok{$}\NormalTok{NCBI ,}\FunctionTok{paste}\NormalTok{(y}\SpecialCharTok{$}\NormalTok{NCBI,}\StringTok{"."}\NormalTok{,y}\SpecialCharTok{$}\NormalTok{init\_final, }\AttributeTok{sep =} \StringTok{""}\NormalTok{) )}
\NormalTok{y}\SpecialCharTok{$}\NormalTok{init\_final}\OtherTok{\textless{}{-}}\FunctionTok{ifelse}\NormalTok{(}\FunctionTok{str\_detect}\NormalTok{(y}\SpecialCharTok{$}\NormalTok{init\_final, }\StringTok{"{-}"}\NormalTok{), y}\SpecialCharTok{$}\NormalTok{init\_final, y}\SpecialCharTok{$}\StringTok{\textasciigrave{}}\AttributeTok{Model Sequence ID}\StringTok{\textasciigrave{}}\NormalTok{)}
\NormalTok{y}\SpecialCharTok{$}\StringTok{\textasciigrave{}}\AttributeTok{Model Sequence ID}\StringTok{\textasciigrave{}}\OtherTok{\textless{}{-}}\FunctionTok{ifelse}\NormalTok{(}\FunctionTok{str\_detect}\NormalTok{(y}\SpecialCharTok{$}\StringTok{\textasciigrave{}}\AttributeTok{Model Sequence ID}\StringTok{\textasciigrave{}}\NormalTok{, }\StringTok{"{-}"}\NormalTok{), y}\SpecialCharTok{$}\NormalTok{X, y}\SpecialCharTok{$}\StringTok{\textasciigrave{}}\AttributeTok{Model Sequence ID}\StringTok{\textasciigrave{}}\NormalTok{)}
\NormalTok{y}\SpecialCharTok{$}\StringTok{\textasciigrave{}}\AttributeTok{ARO Accession}\StringTok{\textasciigrave{}}\OtherTok{\textless{}{-}}\FunctionTok{ifelse}\NormalTok{(y}\SpecialCharTok{$}\StringTok{\textasciigrave{}}\AttributeTok{ARO Accession}\StringTok{\textasciigrave{}}\SpecialCharTok{\textless{}}\DecValTok{3000000}\NormalTok{, y}\SpecialCharTok{$}\NormalTok{NCBI, y}\SpecialCharTok{$}\StringTok{\textasciigrave{}}\AttributeTok{ARO Accession}\StringTok{\textasciigrave{}}\NormalTok{)}
\NormalTok{y}\SpecialCharTok{$}\StringTok{\textasciigrave{}}\AttributeTok{ARO Accession}\StringTok{\textasciigrave{}}\OtherTok{\textless{}{-}}\FunctionTok{str\_split\_fixed}\NormalTok{(y}\SpecialCharTok{$}\StringTok{\textasciigrave{}}\AttributeTok{ARO Accession}\StringTok{\textasciigrave{}}\NormalTok{, }\StringTok{"}\SpecialCharTok{\textbackslash{}\textbackslash{}}\StringTok{."}\NormalTok{, }\DecValTok{2}\NormalTok{)[ ,}\DecValTok{1}\NormalTok{]}
\NormalTok{y}\OtherTok{\textless{}{-}}\FunctionTok{select}\NormalTok{(y, }\SpecialCharTok{{-}}\NormalTok{X)}
\NormalTok{Groot\_data}\OtherTok{\textless{}{-}}\FunctionTok{cbind}\NormalTok{(Groot\_data,y)}
\NormalTok{Groot\_data}\SpecialCharTok{$}\StringTok{\textasciigrave{}}\AttributeTok{ARO Accession}\StringTok{\textasciigrave{}}\OtherTok{\textless{}{-}}\FunctionTok{as.numeric}\NormalTok{(Groot\_data}\SpecialCharTok{$}\StringTok{\textasciigrave{}}\AttributeTok{ARO Accession}\StringTok{\textasciigrave{}}\NormalTok{)}
\NormalTok{Groot\_data}\OtherTok{\textless{}{-}}\FunctionTok{merge}\NormalTok{(Groot\_data, }\FunctionTok{select}\NormalTok{(aro\_index,}\StringTok{\textasciigrave{}}\AttributeTok{Model Sequence ID}\StringTok{\textasciigrave{}}\NormalTok{, }\StringTok{\textasciigrave{}}\AttributeTok{AMR Gene Family}\StringTok{\textasciigrave{}}\NormalTok{,}\StringTok{\textasciigrave{}}\AttributeTok{Drug Class}\StringTok{\textasciigrave{}}\NormalTok{, }\StringTok{\textasciigrave{}}\AttributeTok{Resistance Mechanism}\StringTok{\textasciigrave{}}\NormalTok{,}\StringTok{\textasciigrave{}}\AttributeTok{Protein Accession}\StringTok{\textasciigrave{}}\NormalTok{, }\StringTok{\textasciigrave{}}\AttributeTok{ARO Number}\StringTok{\textasciigrave{}}\NormalTok{, }\StringTok{\textasciigrave{}}\AttributeTok{ARO Name}\StringTok{\textasciigrave{}}\NormalTok{) ,  }\AttributeTok{by.x =} \StringTok{"ARO Accession"}\NormalTok{, }\AttributeTok{by.y=}\StringTok{"ARO Number"}\NormalTok{, }\AttributeTok{all.x =} \ConstantTok{TRUE}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Remove non coding AMR sequences:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{Groot\_data}\OtherTok{\textless{}{-}}\NormalTok{Groot\_data}\SpecialCharTok{\%\textgreater{}\%}\FunctionTok{filter}\NormalTok{(}\SpecialCharTok{!}\FunctionTok{is.na}\NormalTok{(}\StringTok{\textasciigrave{}}\AttributeTok{Protein Accession}\StringTok{\textasciigrave{}}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

Calculate RPKM:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{Groot\_data}\OtherTok{\textless{}{-}}\NormalTok{Groot\_data}\SpecialCharTok{\%\textgreater{}\%}\FunctionTok{mutate}\NormalTok{(}\AttributeTok{RPKM=}\NormalTok{read.count }\SpecialCharTok{/}\NormalTok{(gene.length}\SpecialCharTok{/}\DecValTok{1000} \SpecialCharTok{*}\NormalTok{ totalNumReads}\SpecialCharTok{/}\DecValTok{1000000}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(knitr)}
\FunctionTok{n\_distinct}\NormalTok{(Groot\_data}\SpecialCharTok{$}\StringTok{\textasciigrave{}}\AttributeTok{AMR Gene Family}\StringTok{\textasciigrave{}}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 138
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{AMR\_family\_Groot}\OtherTok{\textless{}{-}}\NormalTok{Groot\_data}\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{group\_by}\NormalTok{(}\StringTok{\textasciigrave{}}\AttributeTok{AMR Gene Family}\StringTok{\textasciigrave{}}\NormalTok{)}\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{summarise}\NormalTok{(}\StringTok{\textquotesingle{}number of AMR genes per family (groot)\textquotesingle{}}\OtherTok{=}\FunctionTok{n}\NormalTok{(),}
            \AttributeTok{sum\_RPKM\_groot=}\FunctionTok{sum}\NormalTok{(RPKM))}\SpecialCharTok{\%\textgreater{}\%}\FunctionTok{arrange}\NormalTok{(}\FunctionTok{desc}\NormalTok{(sum\_RPKM\_groot))}
\FunctionTok{kable}\NormalTok{(}\FunctionTok{head}\NormalTok{(AMR\_family\_Groot))}
\end{Highlighting}
\end{Shaded}

\begin{longtable}[]{@{}lrr@{}}
\toprule
AMR Gene Family & number of AMR genes per family (groot) &
sum\_RPKM\_groot \\
\midrule
\endhead
OXA beta-lactamase & 286 & 127376.53 \\
TEM beta-lactamase & 147 & 96640.76 \\
SHV beta-lactamase & 113 & 86700.04 \\
CTX-M beta-lactamase & 125 & 71126.97 \\
CMY beta-lactamase & 103 & 60814.02 \\
quinolone resistance protein (qnr) & 83 & 43595.37 \\
\bottomrule
\end{longtable}

\hypertarget{ariba-vs-rgi-vs-groot}{%
\section{ARIBA vs RGI vs GROOT}\label{ariba-vs-rgi-vs-groot}}

Number of AMR Gene Families detected by each pipeline:

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{length}\NormalTok{(AMR\_family\_ARIBA}\SpecialCharTok{$}\StringTok{\textasciigrave{}}\AttributeTok{AMR Gene Family}\StringTok{\textasciigrave{}}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 97
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{length}\NormalTok{(AMR\_family\_RGI}\SpecialCharTok{$}\StringTok{\textasciigrave{}}\AttributeTok{AMR Gene Family}\StringTok{\textasciigrave{}}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 125
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{length}\NormalTok{(AMR\_family\_Groot}\SpecialCharTok{$}\StringTok{\textasciigrave{}}\AttributeTok{AMR Gene Family}\StringTok{\textasciigrave{}}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 138
\end{verbatim}

Merge data obtained from the 3 pipelines:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{AMR\_family}\OtherTok{\textless{}{-}}\FunctionTok{merge}\NormalTok{(AMR\_family\_ARIBA, AMR\_family\_RGI[ ,}\DecValTok{1}\SpecialCharTok{:}\DecValTok{3}\NormalTok{], }\AttributeTok{by=}\StringTok{"AMR Gene Family"}\NormalTok{, }\AttributeTok{all=}\ConstantTok{TRUE}\NormalTok{)}
\NormalTok{AMR\_family}\OtherTok{\textless{}{-}}\FunctionTok{merge}\NormalTok{(AMR\_family, AMR\_family\_Groot, }\AttributeTok{by=}\StringTok{"AMR Gene Family"}\NormalTok{, }\AttributeTok{all=}\ConstantTok{TRUE}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{venntransform}\OtherTok{\textless{}{-}}\ControlFlowTok{function}\NormalTok{(x)\{}
\NormalTok{  x}\SpecialCharTok{$}\NormalTok{RGI}\OtherTok{\textless{}{-}}\FunctionTok{ifelse}\NormalTok{(}\FunctionTok{is.na}\NormalTok{(x}\SpecialCharTok{$}\NormalTok{sum\_RPKM\_RGI),}\ConstantTok{FALSE}\NormalTok{,}\ConstantTok{TRUE}\NormalTok{)}
\NormalTok{  x}\SpecialCharTok{$}\NormalTok{Ariba}\OtherTok{\textless{}{-}}\FunctionTok{ifelse}\NormalTok{(}\FunctionTok{is.na}\NormalTok{(x}\SpecialCharTok{$}\NormalTok{sum\_RPKM\_ariba),}\ConstantTok{FALSE}\NormalTok{,}\ConstantTok{TRUE}\NormalTok{)}
\NormalTok{  x}\SpecialCharTok{$}\NormalTok{Groot}\OtherTok{\textless{}{-}}\FunctionTok{ifelse}\NormalTok{(}\FunctionTok{is.na}\NormalTok{(x}\SpecialCharTok{$}\NormalTok{sum\_RPKM\_groot),}\ConstantTok{FALSE}\NormalTok{,}\ConstantTok{TRUE}\NormalTok{)}
\NormalTok{  x}\SpecialCharTok{$}\NormalTok{All}\OtherTok{\textless{}{-}}\FunctionTok{ifelse}\NormalTok{(x}\SpecialCharTok{$}\NormalTok{RGI}\SpecialCharTok{==}\ConstantTok{TRUE} \SpecialCharTok{\&}\NormalTok{ x}\SpecialCharTok{$}\NormalTok{Ariba}\SpecialCharTok{==}\ConstantTok{TRUE} \SpecialCharTok{\&}\NormalTok{ x}\SpecialCharTok{$}\NormalTok{Groot}\SpecialCharTok{==}\ConstantTok{TRUE}\NormalTok{, }\ConstantTok{TRUE}\NormalTok{, }\ConstantTok{FALSE}\NormalTok{)}
  \FunctionTok{return}\NormalTok{(x)\}}

\FunctionTok{library}\NormalTok{(ggvenn)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Loading required package: grid
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{vennplot}\OtherTok{\textless{}{-}}\ControlFlowTok{function}\NormalTok{(x)\{}
\NormalTok{y}\OtherTok{\textless{}{-}}\FunctionTok{ggplot}\NormalTok{(x) }\SpecialCharTok{+} \FunctionTok{geom\_venn}\NormalTok{(}\FunctionTok{aes}\NormalTok{(}\AttributeTok{A =}\NormalTok{ RGI, }\AttributeTok{B =}\NormalTok{ Ariba, }\AttributeTok{C =}\NormalTok{ Groot),}\AttributeTok{fill\_color =} \FunctionTok{c}\NormalTok{(}\StringTok{"chartreuse"}\NormalTok{, }\StringTok{"orange"}\NormalTok{, }\StringTok{"darkorchid1"}\NormalTok{), }\AttributeTok{text\_size=}\DecValTok{4}\NormalTok{, }\AttributeTok{set\_name\_size =} \DecValTok{2}\NormalTok{, }\AttributeTok{stroke\_size =} \FloatTok{0.2}\NormalTok{, }\AttributeTok{show\_percentage =} \ConstantTok{FALSE}\NormalTok{)}\SpecialCharTok{+}\FunctionTok{theme\_void}\NormalTok{()}\SpecialCharTok{+}\FunctionTok{coord\_fixed}\NormalTok{()}\SpecialCharTok{+}\FunctionTok{theme}\NormalTok{(}\AttributeTok{title=}\FunctionTok{element\_text}\NormalTok{(}\AttributeTok{size=}\DecValTok{5}\NormalTok{))}
  \FunctionTok{return}\NormalTok{(y)\}}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{AMR\_family\_venn}\OtherTok{\textless{}{-}}\FunctionTok{venntransform}\NormalTok{(AMR\_family)}
\FunctionTok{vennplot}\NormalTok{(AMR\_family\_venn)}
\end{Highlighting}
\end{Shaded}

\includegraphics{test_data_rpkm_files/figure-latex/unnamed-chunk-26-1.pdf}

Transform data according to which pipeline has detected each AMR gene
family:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{data\_trans}\OtherTok{\textless{}{-}}\ControlFlowTok{function}\NormalTok{(x)\{}
\NormalTok{x}\SpecialCharTok{$}\NormalTok{RGI}\OtherTok{\textless{}{-}}\FunctionTok{ifelse}\NormalTok{(}\FunctionTok{is.na}\NormalTok{(x}\SpecialCharTok{$}\NormalTok{sum\_RPKM\_RGI),}\DecValTok{0}\NormalTok{,}\DecValTok{1}\NormalTok{)}
\NormalTok{x}\SpecialCharTok{$}\NormalTok{Ariba}\OtherTok{\textless{}{-}}\FunctionTok{ifelse}\NormalTok{(}\FunctionTok{is.na}\NormalTok{(x}\SpecialCharTok{$}\NormalTok{sum\_RPKM\_ariba),}\DecValTok{0}\NormalTok{,}\DecValTok{1}\NormalTok{)}
\NormalTok{x}\SpecialCharTok{$}\NormalTok{Groot}\OtherTok{\textless{}{-}}\FunctionTok{ifelse}\NormalTok{(}\FunctionTok{is.na}\NormalTok{(x}\SpecialCharTok{$}\NormalTok{sum\_RPKM\_groot),}\DecValTok{0}\NormalTok{,}\DecValTok{1}\NormalTok{)}
\NormalTok{x}\SpecialCharTok{$}\NormalTok{npipelines}\OtherTok{\textless{}{-}}\FunctionTok{apply}\NormalTok{(}\FunctionTok{select}\NormalTok{(x, RGI, Ariba,Groot),}\DecValTok{1}\NormalTok{, sum)}
\NormalTok{x}\SpecialCharTok{$}\NormalTok{npipelines}\OtherTok{\textless{}{-}}\FunctionTok{as.factor}\NormalTok{(x}\SpecialCharTok{$}\NormalTok{npipelines)}
\NormalTok{x}\SpecialCharTok{$}\NormalTok{pipelines}\OtherTok{\textless{}{-}}\FunctionTok{ifelse}\NormalTok{(x}\SpecialCharTok{$}\NormalTok{npipelines}\SpecialCharTok{==}\DecValTok{3}\NormalTok{, }\StringTok{"All"}\NormalTok{, }
                                \FunctionTok{ifelse}\NormalTok{(x}\SpecialCharTok{$}\NormalTok{npipelines}\SpecialCharTok{==}\DecValTok{2} \SpecialCharTok{\&}\NormalTok{ x}\SpecialCharTok{$}\NormalTok{RGI}\SpecialCharTok{==}\DecValTok{0}\NormalTok{, }\StringTok{"Groot+Ariba"}\NormalTok{,}
                                       \FunctionTok{ifelse}\NormalTok{(x}\SpecialCharTok{$}\NormalTok{npipelines}\SpecialCharTok{==}\DecValTok{2} \SpecialCharTok{\&}\NormalTok{ x}\SpecialCharTok{$}\NormalTok{Ariba}\SpecialCharTok{==}\DecValTok{0}\NormalTok{, }\StringTok{"RGI+Groot"}\NormalTok{,}
                                              \FunctionTok{ifelse}\NormalTok{(x}\SpecialCharTok{$}\NormalTok{npipelines}\SpecialCharTok{==}\DecValTok{2} \SpecialCharTok{\&}\NormalTok{ x}\SpecialCharTok{$}\NormalTok{Groot}\SpecialCharTok{==}\DecValTok{0}\NormalTok{, }\StringTok{"RGI+Ariba"}\NormalTok{,}
                                                     \FunctionTok{ifelse}\NormalTok{(x}\SpecialCharTok{$}\NormalTok{RGI}\SpecialCharTok{==}\DecValTok{1}\NormalTok{, }\StringTok{"RGI only"}\NormalTok{,}
                                                            \FunctionTok{ifelse}\NormalTok{(x}\SpecialCharTok{$}\NormalTok{Ariba}\SpecialCharTok{==}\DecValTok{1}\NormalTok{, }\StringTok{"Ariba only"}\NormalTok{, }\StringTok{"Groot only"}\NormalTok{))))))}
\NormalTok{x}\SpecialCharTok{$}\NormalTok{pipelines\_class}\OtherTok{\textless{}{-}}\FunctionTok{ifelse}\NormalTok{(x}\SpecialCharTok{$}\NormalTok{npipelines}\SpecialCharTok{==}\DecValTok{1}\NormalTok{,}\DecValTok{0}\NormalTok{,}\DecValTok{1}\NormalTok{)}
\FunctionTok{return}\NormalTok{(x)\}}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{AMR\_family}\OtherTok{\textless{}{-}}\FunctionTok{data\_trans}\NormalTok{(AMR\_family)}
\end{Highlighting}
\end{Shaded}

Store discordant families names:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{discrodant\_families}\OtherTok{\textless{}{-}}\FunctionTok{list}\NormalTok{(AMR\_family[AMR\_family}\SpecialCharTok{$}\NormalTok{pipelines}\SpecialCharTok{==}\StringTok{"RGI only"}\NormalTok{ ,}\DecValTok{1}\NormalTok{],}
\NormalTok{AMR\_family[AMR\_family}\SpecialCharTok{$}\NormalTok{pipelines}\SpecialCharTok{==}\StringTok{"Groot only"}\NormalTok{ ,}\DecValTok{1}\NormalTok{],}
\NormalTok{AMR\_family[AMR\_family}\SpecialCharTok{$}\NormalTok{pipelines}\SpecialCharTok{==}\StringTok{"RGI+Groot"}\NormalTok{ ,}\DecValTok{1}\NormalTok{],}
\NormalTok{AMR\_family[AMR\_family}\SpecialCharTok{$}\NormalTok{pipelines}\SpecialCharTok{==}\StringTok{"Groot+Ariba"}\NormalTok{ ,}\DecValTok{1}\NormalTok{])}
\FunctionTok{names}\NormalTok{(discrodant\_families)}\OtherTok{\textless{}{-}}\FunctionTok{c}\NormalTok{(}\StringTok{"RGI only"}\NormalTok{, }\StringTok{"Groot only"}\NormalTok{, }\StringTok{"RGI+Groot"}\NormalTok{, }\StringTok{"Groot+Ariba"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{capture.output}\NormalTok{(discrodant\_families, }\AttributeTok{file =} \StringTok{"discrodant\_families.txt"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

RPKM distribution of AMR gene families according to the number of
pipelines:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{AMR\_family}\SpecialCharTok{$}\NormalTok{pipelines}\OtherTok{\textless{}{-}}\FunctionTok{factor}\NormalTok{(AMR\_family}\SpecialCharTok{$}\NormalTok{pipelines, }\AttributeTok{ordered =} \ConstantTok{TRUE}\NormalTok{, }\AttributeTok{levels =} \FunctionTok{c}\NormalTok{(}\StringTok{"RGI only"}\NormalTok{, }\StringTok{"Ariba only"}\NormalTok{, }\StringTok{"Groot only"}\NormalTok{, }\StringTok{"Groot+Ariba"}\NormalTok{, }\StringTok{"RGI+Groot"}\NormalTok{, }\StringTok{"RGI+Ariba"}\NormalTok{, }\StringTok{"All"}\NormalTok{))}

\NormalTok{p\_RGI}\OtherTok{\textless{}{-}}\FunctionTok{ggplot}\NormalTok{(}\FunctionTok{subset}\NormalTok{(AMR\_family, AMR\_family}\SpecialCharTok{$}\NormalTok{RGI}\SpecialCharTok{==}\DecValTok{1}\NormalTok{), }\FunctionTok{aes}\NormalTok{(pipelines, sum\_RPKM\_RGI))}\SpecialCharTok{+}\FunctionTok{geom\_boxplot}\NormalTok{(}\FunctionTok{aes}\NormalTok{(}\AttributeTok{colour=}\NormalTok{pipelines), }\AttributeTok{outlier.shape =} \ConstantTok{NA}\NormalTok{, }\AttributeTok{na.rm =} \ConstantTok{TRUE}\NormalTok{)}\SpecialCharTok{+}\FunctionTok{geom\_jitter}\NormalTok{(}\AttributeTok{width =} \FloatTok{0.2}\NormalTok{, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{colour=}\NormalTok{pipelines))}\SpecialCharTok{+}\FunctionTok{theme\_classic}\NormalTok{()}\SpecialCharTok{+}\FunctionTok{theme}\NormalTok{(}\AttributeTok{legend.position =} \StringTok{"none"}\NormalTok{)}\SpecialCharTok{+}\FunctionTok{ggtitle}\NormalTok{(}\StringTok{"RGI"}\NormalTok{)}\SpecialCharTok{+}\FunctionTok{scale\_y\_continuous}\NormalTok{(}\AttributeTok{trans=}\StringTok{\textquotesingle{}log10\textquotesingle{}}\NormalTok{)}\SpecialCharTok{+}\FunctionTok{theme}\NormalTok{(}\AttributeTok{axis.text.x =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{angle=}\DecValTok{45}\NormalTok{, }\AttributeTok{hjust =} \DecValTok{1}\NormalTok{), }\AttributeTok{axis.title.x =} \FunctionTok{element\_blank}\NormalTok{())}

\NormalTok{p\_Ariba}\OtherTok{\textless{}{-}}\FunctionTok{ggplot}\NormalTok{(}\FunctionTok{subset}\NormalTok{(AMR\_family, AMR\_family}\SpecialCharTok{$}\NormalTok{Ariba}\SpecialCharTok{==}\DecValTok{1}\NormalTok{), }\FunctionTok{aes}\NormalTok{(pipelines, sum\_RPKM\_ariba))}\SpecialCharTok{+}\FunctionTok{geom\_boxplot}\NormalTok{(}\FunctionTok{aes}\NormalTok{(}\AttributeTok{colour=}\NormalTok{pipelines), }\AttributeTok{outlier.shape =} \ConstantTok{NA}\NormalTok{, }\AttributeTok{na.rm =} \ConstantTok{TRUE}\NormalTok{)}\SpecialCharTok{+}\FunctionTok{geom\_jitter}\NormalTok{(}\AttributeTok{width =} \FloatTok{0.2}\NormalTok{, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{colour=}\NormalTok{pipelines))}\SpecialCharTok{+}\FunctionTok{theme\_classic}\NormalTok{()}\SpecialCharTok{+}\FunctionTok{theme}\NormalTok{(}\AttributeTok{legend.position =} \StringTok{"none"}\NormalTok{)}\SpecialCharTok{+}\FunctionTok{ggtitle}\NormalTok{(}\StringTok{"Ariba"}\NormalTok{)}\SpecialCharTok{+}\FunctionTok{scale\_y\_continuous}\NormalTok{(}\AttributeTok{trans=}\StringTok{\textquotesingle{}log10\textquotesingle{}}\NormalTok{)}\SpecialCharTok{+}\FunctionTok{theme}\NormalTok{(}\AttributeTok{axis.text.x =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{angle=}\DecValTok{45}\NormalTok{, }\AttributeTok{hjust=}\DecValTok{1}\NormalTok{), }\AttributeTok{axis.title.x =} \FunctionTok{element\_blank}\NormalTok{())}

\NormalTok{p\_Groot}\OtherTok{\textless{}{-}}\FunctionTok{ggplot}\NormalTok{(}\FunctionTok{subset}\NormalTok{(AMR\_family, AMR\_family}\SpecialCharTok{$}\NormalTok{Groot}\SpecialCharTok{==}\DecValTok{1}\NormalTok{), }\FunctionTok{aes}\NormalTok{(pipelines, sum\_RPKM\_groot))}\SpecialCharTok{+}\FunctionTok{geom\_boxplot}\NormalTok{(}\FunctionTok{aes}\NormalTok{(}\AttributeTok{colour=}\NormalTok{pipelines), }\AttributeTok{outlier.shape =} \ConstantTok{NA}\NormalTok{, }\AttributeTok{na.rm =} \ConstantTok{TRUE}\NormalTok{)}\SpecialCharTok{+}\FunctionTok{geom\_jitter}\NormalTok{(}\AttributeTok{width =} \FloatTok{0.2}\NormalTok{, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{colour=}\NormalTok{pipelines))}\SpecialCharTok{+}\FunctionTok{theme\_classic}\NormalTok{()}\SpecialCharTok{+}\FunctionTok{theme}\NormalTok{(}\AttributeTok{legend.position =} \StringTok{"none"}\NormalTok{)}\SpecialCharTok{+}\FunctionTok{ggtitle}\NormalTok{(}\StringTok{"Groot"}\NormalTok{)}\SpecialCharTok{+}\FunctionTok{scale\_y\_continuous}\NormalTok{(}\AttributeTok{trans=}\StringTok{\textquotesingle{}log10\textquotesingle{}}\NormalTok{)}\SpecialCharTok{+}\FunctionTok{theme}\NormalTok{(}\AttributeTok{axis.text.x =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{angle=}\DecValTok{45}\NormalTok{, }\AttributeTok{hjust=}\DecValTok{1}\NormalTok{),}\AttributeTok{axis.title.x =} \FunctionTok{element\_blank}\NormalTok{())}
\NormalTok{p\_RGI}\SpecialCharTok{+}\NormalTok{p\_Ariba}\SpecialCharTok{+}\NormalTok{p\_Groot}
\end{Highlighting}
\end{Shaded}

\includegraphics{test_data_rpkm_files/figure-latex/unnamed-chunk-32-1.pdf}

\hypertarget{discordance-analysis}{%
\subsubsection{Discordance analysis:}\label{discordance-analysis}}

The pipelines variable (number of pipelines that have detected an AMR
gene family) is added to each pipeline report:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{RGI\_data}\OtherTok{\textless{}{-}}\FunctionTok{merge}\NormalTok{(RGI\_data, }\FunctionTok{select}\NormalTok{(AMR\_family, pipelines, }\StringTok{\textasciigrave{}}\AttributeTok{AMR Gene Family}\StringTok{\textasciigrave{}}\NormalTok{), }\AttributeTok{by=}\StringTok{"AMR Gene Family"}\NormalTok{, }\AttributeTok{all.x =} \ConstantTok{TRUE}\NormalTok{)}

\NormalTok{Groot\_data}\OtherTok{\textless{}{-}}\FunctionTok{merge}\NormalTok{(Groot\_data, }\FunctionTok{select}\NormalTok{(AMR\_family, pipelines, }\StringTok{\textasciigrave{}}\AttributeTok{AMR Gene Family}\StringTok{\textasciigrave{}}\NormalTok{), }\AttributeTok{by=}\StringTok{"AMR Gene Family"}\NormalTok{, }\AttributeTok{all.x =} \ConstantTok{TRUE}\NormalTok{)}

\NormalTok{Ariba\_data}\OtherTok{\textless{}{-}}\FunctionTok{merge}\NormalTok{(Ariba\_data, }\FunctionTok{select}\NormalTok{(AMR\_family, pipelines, }\StringTok{\textasciigrave{}}\AttributeTok{AMR Gene Family}\StringTok{\textasciigrave{}}\NormalTok{), }\AttributeTok{by=}\StringTok{"AMR Gene Family"}\NormalTok{, }\AttributeTok{all.x =} \ConstantTok{TRUE}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

For each pipeline the discordant sequences are selected:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{RGI\_seqs}\OtherTok{\textless{}{-}}\NormalTok{RGI\_data}\SpecialCharTok{\%\textgreater{}\%}\FunctionTok{filter}\NormalTok{(pipelines}\SpecialCharTok{!=}\StringTok{"All"}\NormalTok{)}\SpecialCharTok{\%\textgreater{}\%}\FunctionTok{select}\NormalTok{(}\StringTok{\textasciigrave{}}\AttributeTok{Reference Sequence}\StringTok{\textasciigrave{}}\NormalTok{)}
\NormalTok{Groot\_seqs}\OtherTok{\textless{}{-}}\NormalTok{Groot\_data}\SpecialCharTok{\%\textgreater{}\%}\FunctionTok{filter}\NormalTok{(pipelines}\SpecialCharTok{!=}\StringTok{"All"}\NormalTok{)}\SpecialCharTok{\%\textgreater{}\%}\FunctionTok{select}\NormalTok{(AMR.gene)}
\NormalTok{Ariba\_seqs}\OtherTok{\textless{}{-}}\NormalTok{Ariba\_data}\SpecialCharTok{\%\textgreater{}\%}\FunctionTok{filter}\NormalTok{(pipelines}\SpecialCharTok{!=}\StringTok{"All"}\NormalTok{)}\SpecialCharTok{\%\textgreater{}\%}\FunctionTok{select}\NormalTok{(ref\_name)}
\end{Highlighting}
\end{Shaded}

The original sequences belonging to discordant AMR gene families for
each pipeline are selected (BASH):

\begin{Shaded}
\begin{Highlighting}[]
\BuiltInTok{cd}\NormalTok{ data\_testing/RGI\_analysis}
\ExtensionTok{samtools}\NormalTok{ view argannot{-}150bp{-}10000{-}reads.fastq\_RGI\_output.temp.bam }\KeywordTok{|} \FunctionTok{grep} \AttributeTok{{-}f}\NormalTok{ RGI\_seqs.txt }\OperatorTok{\textgreater{}}\NormalTok{ RGI\_seqs\_bam.txt}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\BuiltInTok{cd}\NormalTok{ data\_testing/groot\_analysis}
\ExtensionTok{samtools}\NormalTok{ view argannot{-}150bp{-}10000{-}reads{-}uniqseq.bam }\KeywordTok{|} \FunctionTok{grep} \AttributeTok{{-}f}\NormalTok{ Groot\_seqs.txt }\OperatorTok{\textgreater{}}\NormalTok{ Groot\_seqs\_bam.txt}
\end{Highlighting}
\end{Shaded}

Ariba does not generate any BAM file.

Open the generated files:\\
data\_seq=sample sequence name\\
Reference Sequence= Reference sequence name

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(readr)}
\NormalTok{RGI\_seqs\_bam }\OtherTok{\textless{}{-}} \FunctionTok{read\_delim}\NormalTok{(}\StringTok{"data\_testing/RGI\_analysis/RGI\_seqs\_bam.txt"}\NormalTok{, }
    \StringTok{"}\SpecialCharTok{\textbackslash{}t}\StringTok{"}\NormalTok{, }\AttributeTok{escape\_double =} \ConstantTok{FALSE}\NormalTok{, }\AttributeTok{col\_names =} \ConstantTok{FALSE}\NormalTok{, }
    \AttributeTok{trim\_ws =} \ConstantTok{TRUE}\NormalTok{)}
\NormalTok{RGI\_seqs\_bam}\OtherTok{\textless{}{-}}\NormalTok{RGI\_seqs\_bam}\SpecialCharTok{\%\textgreater{}\%}\FunctionTok{select}\NormalTok{(X1, X3)}\SpecialCharTok{\%\textgreater{}\%}\FunctionTok{rename}\NormalTok{(}\AttributeTok{data\_seq=}\NormalTok{X1, }\StringTok{\textasciigrave{}}\AttributeTok{Reference Sequence}\StringTok{\textasciigrave{}}\OtherTok{=}\NormalTok{X3 )}
\FunctionTok{head}\NormalTok{(RGI\_seqs\_bam)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## # A tibble: 6 x 2
##   data_seq                                       `Reference Sequence`           
##   <chr>                                          <chr>                          
## 1 SYN_24_461_610_0_-_2279391_1_._argannot~~~(Ri~ ARO:3002884|ID:796|Name:iri|NC~
## 2 SYN_9_833_982_0_+_336169_1_._argannot~~~(Bla)~ ARO:3002188|ID:53|Name:MOX-5|N~
## 3 SYN_15_592_741_0_-_2223819_1_._argannot~~~(ML~ ARO:3001308|ID:1144|Name:vgbB|~
## 4 SYN_86_130_279_0_-_335466_1_._argannot~~~(Bla~ ARO:3003553|ID:299|Name:CepS|N~
## 5 SYN_89_122_271_0_-_932537_1_._argannot~~~(Bla~ ARO:3002188|ID:53|Name:MOX-5|N~
## 6 SYN_122_576_725_0_+_302036_1_._argannot~~~(Bl~ ARO:3004755|ID:3562|Name:BIL-1~
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{Groot\_seqs\_bam }\OtherTok{\textless{}{-}} \FunctionTok{read\_delim}\NormalTok{(}\StringTok{"data\_testing/groot\_analysis/Groot\_seqs\_bam.txt"}\NormalTok{, }
    \StringTok{"}\SpecialCharTok{\textbackslash{}t}\StringTok{"}\NormalTok{, }\AttributeTok{escape\_double =} \ConstantTok{FALSE}\NormalTok{, }\AttributeTok{col\_names =} \ConstantTok{FALSE}\NormalTok{, }
    \AttributeTok{trim\_ws =} \ConstantTok{TRUE}\NormalTok{)}
\NormalTok{Groot\_seqs\_bam}\OtherTok{\textless{}{-}}\NormalTok{Groot\_seqs\_bam}\SpecialCharTok{\%\textgreater{}\%}\FunctionTok{select}\NormalTok{(X1, X3)}\SpecialCharTok{\%\textgreater{}\%}\FunctionTok{rename}\NormalTok{(}\AttributeTok{data\_seq=}\NormalTok{X1, }\StringTok{\textasciigrave{}}\AttributeTok{Reference Sequence}\StringTok{\textasciigrave{}}\OtherTok{=}\NormalTok{X3 )}
\NormalTok{Groot\_seqs\_bam}\SpecialCharTok{$}\StringTok{\textasciigrave{}}\AttributeTok{Reference Sequence}\StringTok{\textasciigrave{}}\OtherTok{\textless{}{-}}\FunctionTok{str\_remove}\NormalTok{(Groot\_seqs\_bam}\SpecialCharTok{$}\StringTok{\textasciigrave{}}\AttributeTok{Reference Sequence}\StringTok{\textasciigrave{}}\NormalTok{, }\StringTok{"}\SpecialCharTok{\textbackslash{}\textbackslash{}}\StringTok{*"}\NormalTok{)}
\FunctionTok{head}\NormalTok{(Groot\_seqs\_bam)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## # A tibble: 6 x 2
##   data_seq                                           `Reference Sequence`       
##   <chr>                                              <chr>                      
## 1 SYN_24_461_610_0_-_2279391_1_._argannot~~~(Rif)IR~ iri.3002884.U56415.279-171~
## 2 SYN_13_456_605_0_-_894076_1_._argannot~~~(Bla)LEN~ LEN-18.3002465.AM850908.0-~
## 3 SYN_89_122_271_0_-_932537_1_._argannot~~~(Bla)MOX~ MOX-8.3002190.JX173956.0-1~
## 4 SYN_15_592_741_0_-_2223819_1_._argannot~~~(MLS)Vg~ vgbB.3001308.AF015628.398-~
## 5 SYN_161_46_195_0_+_877304_1_._argannot~~~(Bla)LEN~ LEN-16.3002464.AY743416.67~
## 6 SYN_239_182_331_0_+_955099_1_._argannot~~~(Bla)ND~ NDM-6.3002356.JN967644.0-8~
\end{verbatim}

Merge data to obtain CARD information from Reference sequences and
extract a simpler version of the sample sequence names. Compare sample
sequence names to ARO Term names from Reference sequences.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{RGI\_seqs\_bam}\OtherTok{\textless{}{-}}\FunctionTok{merge}\NormalTok{(RGI\_seqs\_bam, }\FunctionTok{select}\NormalTok{(RGI\_data, }\StringTok{\textasciigrave{}}\AttributeTok{AMR Gene Family}\StringTok{\textasciigrave{}}\NormalTok{,}\StringTok{\textasciigrave{}}\AttributeTok{Reference Sequence}\StringTok{\textasciigrave{}}\NormalTok{,}\StringTok{\textasciigrave{}}\AttributeTok{ARO Term}\StringTok{\textasciigrave{}}\NormalTok{, }\StringTok{\textasciigrave{}}\AttributeTok{Resistance Mechanism}\StringTok{\textasciigrave{}}\NormalTok{, }\StringTok{\textasciigrave{}}\AttributeTok{Drug Class}\StringTok{\textasciigrave{}}\NormalTok{, RPKM, pipelines), }\AttributeTok{by=}\StringTok{"Reference Sequence"}\NormalTok{, }\AttributeTok{all.x =} \ConstantTok{TRUE}\NormalTok{)}
\NormalTok{RGI\_seqs\_bam}\SpecialCharTok{$}\NormalTok{data\_seq\_simplified}\OtherTok{\textless{}{-}}\FunctionTok{str\_split\_fixed}\NormalTok{(RGI\_seqs\_bam}\SpecialCharTok{$}\NormalTok{data\_seq, }\StringTok{"\textasciitilde{}\textasciitilde{}\textasciitilde{}"}\NormalTok{, }\DecValTok{3}\NormalTok{)[,}\DecValTok{2}\NormalTok{]}
\NormalTok{RGI\_seqs\_bam}\SpecialCharTok{$}\NormalTok{data\_seq\_simplified}\OtherTok{\textless{}{-}}\FunctionTok{str\_split\_fixed}\NormalTok{(RGI\_seqs\_bam}\SpecialCharTok{$}\NormalTok{data\_seq\_simplified, }\StringTok{"}\SpecialCharTok{\textbackslash{}\textbackslash{}}\StringTok{)"}\NormalTok{, }\DecValTok{2}\NormalTok{)[ ,}\DecValTok{2}\NormalTok{]}
\NormalTok{RGI\_seqs\_bam}\SpecialCharTok{$}\NormalTok{concordance}\OtherTok{\textless{}{-}}\FunctionTok{ifelse}\NormalTok{(}\FunctionTok{tolower}\NormalTok{(RGI\_seqs\_bam}\SpecialCharTok{$}\NormalTok{data\_seq\_simplified)}\SpecialCharTok{\%in\%}\FunctionTok{tolower}\NormalTok{(RGI\_seqs\_bam}\SpecialCharTok{$}\StringTok{\textasciigrave{}}\AttributeTok{ARO Term}\StringTok{\textasciigrave{}}\NormalTok{) }\SpecialCharTok{|} \FunctionTok{tolower}\NormalTok{(RGI\_seqs\_bam}\SpecialCharTok{$}\StringTok{\textasciigrave{}}\AttributeTok{ARO Term}\StringTok{\textasciigrave{}}\NormalTok{)}\SpecialCharTok{\%in\%}\FunctionTok{tolower}\NormalTok{(RGI\_seqs\_bam}\SpecialCharTok{$}\NormalTok{data\_seq\_simplified),}\DecValTok{1}\NormalTok{,}\DecValTok{0}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{Groot\_seqs\_bam}\OtherTok{\textless{}{-}}\FunctionTok{merge}\NormalTok{(Groot\_seqs\_bam, }\FunctionTok{select}\NormalTok{(Groot\_data, }\StringTok{\textasciigrave{}}\AttributeTok{AMR Gene Family}\StringTok{\textasciigrave{}}\NormalTok{,AMR.gene,}\StringTok{\textasciigrave{}}\AttributeTok{ARO Term}\StringTok{\textasciigrave{}}\NormalTok{, }\StringTok{\textasciigrave{}}\AttributeTok{Resistance Mechanism}\StringTok{\textasciigrave{}}\NormalTok{, }\StringTok{\textasciigrave{}}\AttributeTok{Drug Class}\StringTok{\textasciigrave{}}\NormalTok{, RPKM, pipelines), }\AttributeTok{by.x=}\StringTok{"Reference Sequence"}\NormalTok{, }\AttributeTok{by.y=}\StringTok{"AMR.gene"}\NormalTok{, }\AttributeTok{all.x =} \ConstantTok{TRUE}\NormalTok{)}
\NormalTok{Groot\_seqs\_bam}\SpecialCharTok{$}\NormalTok{data\_seq\_simplified}\OtherTok{\textless{}{-}}\FunctionTok{str\_split\_fixed}\NormalTok{(Groot\_seqs\_bam}\SpecialCharTok{$}\NormalTok{data\_seq, }\StringTok{"\textasciitilde{}\textasciitilde{}\textasciitilde{}"}\NormalTok{, }\DecValTok{3}\NormalTok{)[,}\DecValTok{2}\NormalTok{]}
\NormalTok{Groot\_seqs\_bam}\SpecialCharTok{$}\NormalTok{data\_seq\_simplified}\OtherTok{\textless{}{-}}\FunctionTok{str\_split\_fixed}\NormalTok{(Groot\_seqs\_bam}\SpecialCharTok{$}\NormalTok{data\_seq\_simplified, }\StringTok{"}\SpecialCharTok{\textbackslash{}\textbackslash{}}\StringTok{)"}\NormalTok{, }\DecValTok{2}\NormalTok{)[ ,}\DecValTok{2}\NormalTok{]}
\NormalTok{Groot\_seqs\_bam}\SpecialCharTok{$}\NormalTok{concordance}\OtherTok{\textless{}{-}}\FunctionTok{ifelse}\NormalTok{(}\FunctionTok{tolower}\NormalTok{(Groot\_seqs\_bam}\SpecialCharTok{$}\NormalTok{data\_seq\_simplified)}\SpecialCharTok{\%in\%}\FunctionTok{tolower}\NormalTok{(Groot\_seqs\_bam}\SpecialCharTok{$}\StringTok{\textasciigrave{}}\AttributeTok{ARO Term}\StringTok{\textasciigrave{}}\NormalTok{) }\SpecialCharTok{|} \FunctionTok{tolower}\NormalTok{(Groot\_seqs\_bam}\SpecialCharTok{$}\StringTok{\textasciigrave{}}\AttributeTok{ARO Term}\StringTok{\textasciigrave{}}\NormalTok{)}\SpecialCharTok{\%in\%}\FunctionTok{tolower}\NormalTok{(Groot\_seqs\_bam}\SpecialCharTok{$}\NormalTok{data\_seq\_simplified) ,}\DecValTok{1}\NormalTok{,}\DecValTok{0}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Manual revision of concordance:

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{write\_tsv}\NormalTok{(RGI\_seqs\_bam, }\StringTok{"data\_testing/RGI\_analysis/RGI\_seqs\_bam.tsv"}\NormalTok{)}
\FunctionTok{write\_tsv}\NormalTok{(Groot\_seqs\_bam, }\StringTok{"data\_testing/groot\_analysis/Groot\_seqs\_bam.tsv"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\hypertarget{sensibility-and-specificity}{%
\subsection{Sensibility and
Specificity}\label{sensibility-and-specificity}}

True positives: concordant sequences (identified by 3 pipelines) +
correctly identified discordant sequences False positives: incorrectly
identified discordant sequences False negatives: not mapped sequences
True negatives: 0, all sequences are resistant genes.

Total: 10000 seq

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{RGI\_seqs\_bam}\OtherTok{\textless{}{-}} \FunctionTok{read\_delim}\NormalTok{(}\StringTok{"data\_testing/RGI\_analysis/RGI\_seqs\_bam\_manual.csv"}\NormalTok{, }
    \StringTok{"}\SpecialCharTok{\textbackslash{}t}\StringTok{"}\NormalTok{, }\AttributeTok{escape\_double =} \ConstantTok{FALSE}\NormalTok{, }\AttributeTok{trim\_ws =} \ConstantTok{TRUE}\NormalTok{)}
\FunctionTok{nrow}\NormalTok{(RGI\_seqs\_bam)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 232
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{table}\NormalTok{(RGI\_seqs\_bam}\SpecialCharTok{$}\NormalTok{concordance)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
##   0   1 
##   8 224
\end{verbatim}

\hypertarget{rgi-1}{%
\subsubsection{RGI}\label{rgi-1}}

\begin{itemize}
\tightlist
\item
  9473 mapped sequences: (mapping stats)

  \begin{itemize}
  \tightlist
  \item
    232 discordant sequences

    \begin{itemize}
    \tightlist
    \item
      224 correctly identified\\
    \item
      8 incorrectly identified\\
    \end{itemize}
  \item
    9241 concordant sequences (9473-232)\\
  \end{itemize}
\item
  527 not mapped sequences
\end{itemize}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{vp}\OtherTok{\textless{}{-}}\DecValTok{9241}\SpecialCharTok{+}\DecValTok{232}
\NormalTok{fp}\OtherTok{\textless{}{-}}\DecValTok{8}
\NormalTok{fn}\OtherTok{\textless{}{-}}\DecValTok{527}
\NormalTok{vn}\OtherTok{\textless{}{-}}\DecValTok{0}
\NormalTok{sensibilidad\_rgi}\OtherTok{\textless{}{-}}\NormalTok{vp}\SpecialCharTok{*}\DecValTok{100}\SpecialCharTok{/}\NormalTok{(vp}\SpecialCharTok{+}\NormalTok{fn)}
\NormalTok{especificidad\_rgi}\OtherTok{\textless{}{-}}\NormalTok{vn}\SpecialCharTok{*}\DecValTok{100}\SpecialCharTok{/}\NormalTok{(vn}\SpecialCharTok{+}\NormalTok{fp)}
\NormalTok{vpp\_rgi}\OtherTok{\textless{}{-}}\NormalTok{vp}\SpecialCharTok{*}\DecValTok{100}\SpecialCharTok{/}\NormalTok{(vp}\SpecialCharTok{+}\NormalTok{fp)}

\NormalTok{sensibilidad\_rgi}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 94.73
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{especificidad\_rgi}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 0
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{vpp\_rgi}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 99.91562
\end{verbatim}

\hypertarget{groot-1}{%
\subsubsection{GROOT}\label{groot-1}}

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{samtools}\NormalTok{ view data\_testing/groot\_analysis/argannot{-}150bp{-}10000{-}reads{-}uniqseq.bam }\KeywordTok{|} \FunctionTok{wc} \AttributeTok{{-}l}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 8822
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{Groot\_seqs\_bam}\OtherTok{\textless{}{-}} \FunctionTok{read\_delim}\NormalTok{(}\StringTok{"data\_testing/groot\_analysis/Groot\_seqs\_bam\_manual.csv"}\NormalTok{, }
    \StringTok{"}\SpecialCharTok{\textbackslash{}t}\StringTok{"}\NormalTok{, }\AttributeTok{escape\_double =} \ConstantTok{FALSE}\NormalTok{, }\AttributeTok{trim\_ws =} \ConstantTok{TRUE}\NormalTok{)}
\FunctionTok{nrow}\NormalTok{(Groot\_seqs\_bam)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 694
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{table}\NormalTok{(Groot\_seqs\_bam}\SpecialCharTok{$}\NormalTok{concordance)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
##   0   1 
##  14 680
\end{verbatim}

\begin{itemize}
\tightlist
\item
  8822 mapped sequences: (bam file)

  \begin{itemize}
  \tightlist
  \item
    694 discordant sequences

    \begin{itemize}
    \tightlist
    \item
      680 correctly identified\\
    \item
      14 incorrectly identified\\
    \end{itemize}
  \item
    8128 concordant sequences (8822-694)
  \end{itemize}
\item
  1178 not mapped sequences
\end{itemize}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{vp}\OtherTok{\textless{}{-}}\DecValTok{8128}\SpecialCharTok{+}\DecValTok{680}
\NormalTok{fp}\OtherTok{\textless{}{-}}\DecValTok{14}
\NormalTok{fn}\OtherTok{\textless{}{-}}\DecValTok{1178}
\NormalTok{vn}\OtherTok{\textless{}{-}}\DecValTok{0}
\NormalTok{sensibilidad\_groot}\OtherTok{\textless{}{-}}\NormalTok{vp}\SpecialCharTok{*}\DecValTok{100}\SpecialCharTok{/}\NormalTok{(vp}\SpecialCharTok{+}\NormalTok{fn)}
\NormalTok{especificidad\_groot}\OtherTok{\textless{}{-}}\NormalTok{vn}\SpecialCharTok{*}\DecValTok{100}\SpecialCharTok{/}\NormalTok{(vn}\SpecialCharTok{+}\NormalTok{fp)}
\NormalTok{vpp\_groot}\OtherTok{\textless{}{-}}\NormalTok{vp}\SpecialCharTok{*}\DecValTok{100}\SpecialCharTok{/}\NormalTok{(vp}\SpecialCharTok{+}\NormalTok{fp)}

\NormalTok{sensibilidad\_groot}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 88.20348
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{especificidad\_groot}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 0
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{vpp\_groot}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 99.84131
\end{verbatim}

\end{document}
