---
title: "RPKM_multsamples"
author: "Elisa Rubio"
date: "4/27/2021"
output:
  html_document:
    toc: true
    theme: united
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "~/Documentos/Repos/UOC-Master/PRJNA307231")
```

## Import metadata file and nº of reads per sample:

Metadata file:
```{r message=FALSE}
library(knitr)
setwd("~/Documentos/Repos/UOC-Master/PRJNA307231")
library(readr)
metadata <- read_delim("samples_metadata.csv", 
    "\t", escape_double = FALSE, trim_ws = TRUE)
```

Get nº of reads from each sample (from overall_mapping_stats RGI):

```{bash eval=FALSE, include=TRUE}
cd RGI_analysis
for stats in *stats.txt; do
    sample0=${stats%.*}
    sample1=${sample0%.*}
    sample=${sample1#*_}
    totalseq=$(grep ^Total $stats | sed s/[^0-9]//g)
    printf "$sample\t$totalseq\n" >> totalNumReads.csv; done

mv totalNumReads.csv ../
cd ..
```

Merge datasets:
```{r message=FALSE}
library(readr)
totalNumReads <- read_delim("totalNumReads.csv", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)
names(totalNumReads)<-c("Run", "totalNumReads")
metadata<-merge(metadata, totalNumReads, by="Run")
```

## RGI

### Get files (list of dataframes) and assign sample names to each dataframe:
```{r message=FALSE}
RGI_files<-list.files("RGI_analysis", "*.data.txt", full.names = TRUE)
RGI_data <- lapply(RGI_files, function (x) read_delim(x, delim="\t", escape_double = FALSE, trim_ws = TRUE))
names(RGI_data)<-metadata$Run
```

### Calculate RPKM for each dataframe:

```{r}
for (i in 1:length(RGI_data)){
 RGI_data[[i]]$RPKM<-RGI_data[[i]]$`All Mapped Reads`/(RGI_data[[i]]$`Reference Length`/1000 * metadata$totalNumReads[i]/1000000)
}
```

### Plots (MAPQ distribution per sample)

```{r}
library(ggplot2)
library(gridExtra)
plots<-function(x){ggplot(x, aes(`Average MAPQ (Completely Mapped Reads)`)) +
  geom_histogram(bins=50)+theme_bw()}
RGI_plots<-lapply(RGI_data, plots)
grid.arrange(grobs = RGI_plots, ncol = 4)
```

### Remove sequences with MAPQ<10:

```{r}
RGI_data<-lapply(RGI_data, function(x) subset(x, x$`Average MAPQ (Completely Mapped Reads)`>10))
```

### Calculate RPKM sum per AMR gene family:

Create function:

```{r message=FALSE}
library(dplyr)
amrfamily<-function(x, pipeline){
  y<-x%>%group_by(`AMR Gene Family`)%>%summarise(v1=n(), v2=sum(RPKM))
  names(y)[2:3]<-c(paste('nAMR_genes_perfamily_', pipeline, sep=''),paste('sum_RPKM_', pipeline, sep=''))
  return(y)}

drugclass<-function(x, pipeline){
  y<-x%>%group_by(`Drug Class`)%>%summarise(v1=n(), v2=sum(RPKM))
  names(y)[2:3]<-c(paste('nAMR_genes_perdrugclass_', pipeline, sep=''),paste('sum_RPKM_', pipeline, sep=''))
  return(y)}

```

Apply to RGI data:

```{r}
RGI_summary<-lapply(RGI_data, function(x) amrfamily (x, 'RGI'))
RGI_summary2<-lapply(RGI_data, function(x) drugclass (x, 'RGI'))
```


## ARIBA
### Get all report files (list of dataframes) and assign names:
```{r message=FALSE, warning=FALSE}
ariba_files<-list.files("ariba_analysis", full.names = TRUE)
ariba_data <- lapply(ariba_files, function (x) read_delim(x, delim="\t", escape_double = FALSE, trim_ws = TRUE))
names(ariba_data)<-metadata$Run
```
### Remove contig and variant data from ariba reports: 

```{r}
aribasubs<-function(x){x%>%select(c(1:8,31))%>%unique()}
aribasubs2<-function(x){x%>% filter(gene==1)%>%select(c(1:8,31))%>%unique()} ##Only coding sequences
ariba_data<-lapply(ariba_data, aribasubs)
```

### Calculate RPKM (using total number of reads):

```{r}
for (i in 1:length(ariba_data)){
 ariba_data[[i]]$RPKM<-ariba_data[[i]]$`reads`/(ariba_data[[i]]$`ref_len`/1000 * metadata$totalNumReads[i]/1000000)}
```


### Obtain AMR gene family

Import CARD data aro index: 

```{r message=FALSE, warning=FALSE}
aro_index <- read_delim("aro_index.tsv", 
    "\t", escape_double = FALSE, trim_ws = TRUE)
```

Trasform ARIBA notation to merge with CARD DB and obtain AMR gene family and drug class: 
```{r}
library(stringr)
library(reshape2)
aribanotation<-function(x){
y<-colsplit(x$ref_name, "\\.", c("ARO Term", "ARO Accession","NCBI","init_final","Model Sequence ID","v1"))
y$NCBI<-ifelse(str_detect(y$init_final, "_"),y$NCBI ,paste(y$NCBI,".",y$init_final) )
y$init_final<-ifelse(str_detect(y$init_final, "_"), y$init_final, y$`Model Sequence ID`)
y$`Model Sequence ID`<-ifelse(str_detect(y$`Model Sequence ID`, "_"), y$v1, y$`Model Sequence ID`)
y<-select(y, -v1)
x<-cbind(x,y)
x$'ARO Accession'<-paste('ARO:', x$'ARO Accession', sep='')
x<-merge(x, select(aro_index,`Model Sequence ID`, `AMR Gene Family`,`Drug Class`, `Resistance Mechanism`, `ARO Accession`) , by="ARO Accession", all.x=TRUE)
return(x)}
```

Apply to list of dataframes:
```{r}
ariba_data<-lapply(ariba_data, aribanotation)
```

```{r eval=FALSE, include=TRUE}
lapply(ariba_data, function(x) table(is.na(x$`AMR Gene Family`)))
```

### Calculate RPKM sum per AMR gene family and Drug Class: 

```{r}
ariba_summary<-lapply(ariba_data, function(x) amrfamily (x, 'ariba'))
ariba_summary2<-lapply(ariba_data, function(x) drugclass (x, 'ariba'))
```

## GROOT

### Get files (list of dataframes) and assign names:
```{r message=FALSE}
groot_files<-list.files("groot_analysis", "*.0report", full.names = TRUE)
groot_data <- lapply(groot_files, function (x) read_delim(x, delim="\t", escape_double = FALSE, trim_ws = TRUE, col_names = c("AMR.gene", "read.count", "gene.length", "coverage.cigar")))
names(groot_data)<-metadata$Run
```

### Calculate RPKM for each dataframe:

```{r}
for (i in 1:length(groot_data)){
 groot_data[[i]]$RPKM<-groot_data[[i]]$read.count/(groot_data[[i]]$gene.length/1000 * metadata$totalNumReads[i]/1000000)
}
```

### Obtain AMR gene family and Drug Class

Transform GROOT notation to merge with CARD DB and obtain AMR gene family and Drug Class: 

```{r}
grootnotation<-function(x){
y<-colsplit(x$AMR.gene, "\\.", c("ARO Term", "ARO Accession","NCBI","init_final","Model Sequence ID","v1"))
y$NCBI<-ifelse(str_detect(y$init_final, "-"),y$NCBI ,paste(y$NCBI,".",y$init_final) )
y$init_final<-ifelse(str_detect(y$init_final, "-"), y$init_final, y$`Model Sequence ID`)
y$`Model Sequence ID`<-ifelse(str_detect(y$`Model Sequence ID`, "-"), y$v1, y$`Model Sequence ID`)
y<-select(y, -v1)
x<-cbind(x,y)
x$'ARO Accession'<-paste('ARO:', x$'ARO Accession', sep='')
x<-merge(x, select(aro_index,`Model Sequence ID`, `AMR Gene Family`,`Drug Class`, `Resistance Mechanism`, `ARO Accession` ) , by="ARO Accession", all.x=TRUE)
return(x)}
```

Apply to list of dataframes:

```{r}
groot_data<-lapply(groot_data, grootnotation)
```

### Calculate RPKM sum per AMR gene family and Drug Class:

```{r}
groot_summary<-lapply(groot_data, function(x) amrfamily (x, 'groot'))
groot_summary2<-lapply(groot_data, function(x) drugclass (x, 'groot'))
```

## Merge data from the 3 pipelines (grouped by AMR gene family and Drug Class):

```{r}
comparison_list<-Map(merge, RGI_summary, ariba_summary, by="AMR Gene Family", all=TRUE)
comparison_list<-Map(merge, comparison_list, groot_summary, by="AMR Gene Family", all=TRUE)
comparison_df<-do.call(rbind, comparison_list)

comparison_list2<-Map(merge, RGI_summary2, ariba_summary2, by="Drug Class", all=TRUE)
comparison_list2<-Map(merge, comparison_list2, groot_summary2, by="Drug Class", all=TRUE)
comparison_df2<-do.call(rbind, comparison_list2)

```

### Venn diagram

Transform data for Venn diagram (TRUE/FALSE dataframe):
```{r}
venntransform<-function(x){
  x$RGI<-ifelse(is.na(x$sum_RPKM_RGI),FALSE,TRUE)
  x$Ariba<-ifelse(is.na(x$sum_RPKM_ariba),FALSE,TRUE)
  x$Groot<-ifelse(is.na(x$sum_RPKM_groot),FALSE,TRUE)
  x$All<-ifelse(x$RGI==TRUE & x$Ariba==TRUE & x$Groot==TRUE, TRUE, FALSE)
  return(x)}

comparison_list_venn<-lapply(comparison_list, venntransform)
comparison_list2<-lapply(comparison_list2, venntransform)
```

Venn diagram plots:
https://statisticsglobe.com/ggvenn-r-package#basic-information-about-the-ggvenn-package

```{r}
library(ggvenn)
vennplot<-function(x){
y<-ggplot(x) + geom_venn(aes(A = RGI, B = Ariba, C = Groot),fill_color = c("chartreuse", "orange", "darkorchid1"), text_size=4, set_name_size = 2, stroke_size = 0.2, show_percentage = FALSE)+theme_void()+coord_fixed()+theme(title=element_text(size=5))
  return(y)}

venn_plots<-lapply(comparison_list_venn, vennplot)
venn_plots2<-lapply(comparison_list2, vennplot)
for (i in 1:length(venn_plots))
  {venn_plots[[i]]<-venn_plots[[i]]+ggtitle(metadata$Run[i])}
for (i in 1:length(venn_plots2))
  {venn_plots2[[i]]<-venn_plots2[[i]]+ggtitle(metadata$Run[i])}
grid.arrange(grobs = venn_plots, ncol = 4, top="AMR gene family")
grid.arrange(grobs = venn_plots2, ncol = 4, top="Drug Class")
```

```{r}
test<-comparison_list[[1]]
```

### RPKM distribution according to nº of pipelines that identify an AMR gene family: 

```{r}
library(stringr)
comparison_df$Sample<-str_split_fixed(rownames(comparison_df), "\\.", n = 2)[ ,1]
comparison_df2$Sample<-str_split_fixed(rownames(comparison_df2), "\\.", n = 2)[ ,1]


comparison_df$RGI<-ifelse(is.na(comparison_df$sum_RPKM_RGI),0,1)
comparison_df$Ariba<-ifelse(is.na(comparison_df$sum_RPKM_ariba),0,1)
comparison_df$Groot<-ifelse(is.na(comparison_df$sum_RPKM_groot),0,1)
comparison_df$npipelines<-apply(select(comparison_df, RGI, Ariba,Groot),1, sum)
comparison_df$npipelines<-as.factor(comparison_df$npipelines)
comparison_df$pipelines<-ifelse(comparison_df$npipelines==3, "All", 
                                ifelse(comparison_df$npipelines==2 & comparison_df$RGI==0, "Groot+Ariba",
                                       ifelse(comparison_df$npipelines==2 & comparison_df$Ariba==0, "RGI+Groot",
                                              ifelse(comparison_df$npipelines==2 & comparison_df$Groot==0, "RGI+Ariba",
                                                     ifelse(comparison_df$RGI==1, "RGI only",
                                                            ifelse(comparison_df$Ariba==1, "Ariba only", "Groot only"))))))
```


```{r}
ggplot(subset(comparison_df, comparison_df$RGI==1), aes(npipelines, sum_RPKM_RGI))+geom_boxplot(aes(colour=npipelines), outlier.shape = NA)+geom_jitter(width = 0.2, aes(colour=npipelines))+theme_classic()+theme(legend.position = "none")+ggtitle("RGI")+scale_x_discrete(labels=c("1" = "RGI only", "2" = "RGI + other",
                              "3" = "All"))+xlab(NULL)

ggplot(subset(comparison_df, comparison_df$Ariba==1), aes(npipelines, sum_RPKM_ariba))+geom_boxplot(aes(colour=npipelines), outlier.shape = NA)+geom_jitter(width = 0.2, aes(colour=npipelines))+theme_classic()+theme(legend.position = "none")+ggtitle("Ariba")+scale_x_discrete(labels=c("1" = "Ariba only", "2" = "Ariba + other",
                              "3" = "All"))+xlab(NULL)

ggplot(subset(comparison_df, comparison_df$Groot==1), aes(npipelines, sum_RPKM_groot))+geom_boxplot(aes(colour=npipelines), outlier.shape = NA)+geom_jitter(width = 0.2, aes(colour=npipelines))+theme_classic()+theme(legend.position = "none")+ggtitle("Groot")+scale_x_discrete(labels=c("1" = "Groot only", "2" = "Groot + other",
                              "3" = "All"))+xlab(NULL)


```
```{r}
test<-select(comparison_df, 'AMR Gene Family', Sample, sum_RPKM_RGI, sum_RPKM_groot, sum_RPKM_ariba, RGI, Ariba, Groot)
test_RGI<-subset(test, test$RGI==1)
test_Ariba<-subset(test, test$Ariba==1)
test_Groot<-subset(test, test$Groot==1)
```
RGI treshold: 0.1
Ariba no treshold
Groot treshold: 0.01


```{r}
comparison_df$pipelines<-factor(comparison_df$pipelines, ordered = TRUE, levels = c("RGI only", "Ariba only", "Groot only", "Groot+Ariba", "RGI+Groot", "RGI+Ariba", "All"))

p_RGI<-ggplot(subset(comparison_df, comparison_df$RGI==1), aes(pipelines, sum_RPKM_RGI))+geom_boxplot(aes(colour=npipelines), outlier.shape = NA, na.rm = TRUE)+geom_jitter(width = 0.2, aes(colour=npipelines))+theme_classic()+theme(legend.position = "none")+ggtitle("RGI")

p_Ariba<-ggplot(subset(comparison_df, comparison_df$Ariba==1), aes(pipelines, sum_RPKM_ariba))+geom_boxplot(aes(colour=npipelines), outlier.shape = NA, na.rm = TRUE)+geom_jitter(width = 0.2, aes(colour=npipelines))+theme_classic()+theme(legend.position = "none")+ggtitle("Ariba")

p_Groot<-ggplot(subset(comparison_df, comparison_df$Groot==1), aes(pipelines, sum_RPKM_groot))+geom_boxplot(aes(colour=npipelines), outlier.shape = NA, na.rm = TRUE)+geom_jitter(width = 0.2, aes(colour=npipelines))+theme_classic()+theme(legend.position = "none")+ggtitle("Groot")
```

```{r}
p_RGI+ylim(c(0,2))
```

```{r}
p_Ariba ##No treshold
```
```{r}
p_Groot+ylim(c(0,0.2))
```


AMR gene families detected only by one Pipeline
```{r}
RGI_only<-arrange(as.data.frame(table(subset(comparison_df, comparison_df$pipelines=="RGI only")$`AMR Gene Family`)), desc(Freq))
Ariba_only<-arrange(as.data.frame(table(subset(comparison_df, comparison_df$pipelines=="Ariba only")$`AMR Gene Family`)), desc(Freq))
Groot_only<-arrange(as.data.frame(table(subset(comparison_df, comparison_df$pipelines=="Groot only")$`AMR Gene Family`)), desc(Freq))
All<-arrange(as.data.frame(table(subset(comparison_df, comparison_df$pipelines=="All")$`AMR Gene Family`)), desc(Freq))
kable(head(RGI_only))
kable(head(Ariba_only))
kable(head(Groot_only))
kable(head(All))
```

## Apply treshold

```{r}
x<-comparison_list[[1]]
treshold<-function(x){
x$sum_RPKM_RGI<-ifelse(x$sum_RPKM_RGI<=0.1, NA, x$sum_RPKM_RGI)
x$sum_RPKM_groot<-ifelse(x$sum_RPKM_groot<=0.01, NA, x$sum_RPKM_groot)
x$RGI<-ifelse(is.na(x$sum_RPKM_RGI),0,1)
x$Ariba<-ifelse(is.na(x$sum_RPKM_ariba),0,1)
x$Groot<-ifelse(is.na(x$sum_RPKM_groot),0,1)
x$npipelines<-apply(select(x, RGI, Ariba,Groot),1, sum)
x<-subset(x, x$npipelines!=0)}

comparison_list_th<-lapply(comparison_list, treshold)

```

```{r}
comparison_list_th_venn<-lapply(comparison_list_th, venntransform)
venn_plots_th<-lapply(comparison_list_th_venn, vennplot)
for (i in 1:length(venn_plots_th))
{venn_plots_th[[i]]<-venn_plots_th[[i]]+ggtitle(metadata$Run[i])}

grid.arrange(grobs = venn_plots_th, ncol = 4, top="AMR gene family (th)")
```


```{r}
for (i in 1:length(venn_plots))
  {venn_plots[[i]]<-venn_plots[[i]]+ggtitle(metadata$Run[i])}
grid.arrange(grobs = venn_plots, ncol = 4, top="AMR gene family")
```























### Quantitative comparison

```{r}
library(knitr)
kable(head(comparison_df))
kable(head(comparison_df2))
```


```{r}


n_distinct(comparison_df$`AMR Gene Family`)
length(comparison_df$`AMR Gene Family`)

n_distinct(comparison_df2$`Drug Class`)
length(comparison_df2$`Drug Class`)
```
Create a dataframe with all AMR gene families and all Gene classes for each sample:
Change NA values for 0

```{r}
quant_comp<-function(df_comp){
nsamples<-n_distinct(df_comp$Sample)
namrgenes<-n_distinct(df_comp$`AMR Gene Family`)
mat1<-matrix(nrow=namrgenes*nsamples, ncol=2)
mat1[ ,1]<-rep(unique(comparison_df$`AMR Gene Family`), nsamples)
mat1[ ,2]<-rep(metadata$Run, each=namrgenes)
colnames(mat1)<-c('AMR Gene Family', 'Sample')
mat1<-as.data.frame(mat1)
mat2<-merge(mat1, df_comp, by=c("Sample", "AMR Gene Family"), all.x = TRUE)
mat2[is.na(mat2)]<- 0
return(mat2)}
```

```{r}
quant_comp2<-function(df_comp){
nsamples<-n_distinct(df_comp$Sample)
ndrug<-n_distinct(df_comp$`Drug Class`)
mat1<-matrix(nrow=ndrug*nsamples, ncol=2)
mat1[ ,1]<-rep(unique(df_comp$`Drug Class`), nsamples)
mat1[ ,2]<-rep(metadata$Run, each=ndrug)
colnames(mat1)<-c('Drug Class', 'Sample')
mat1<-as.data.frame(mat1)
mat2<-merge(mat1, df_comp, by=c("Sample", "Drug Class"), all.x = TRUE)
mat2[is.na(mat2)]<- 0
return(mat2)}
```

```{r}
comparison_df<-quant_comp(comparison_df)
comparison_df2<-quant_comp2(comparison_df2)
```

Select metadata variables:  
+No more than 1 NA
+At least 2 samples in each group (for categorical variables)

```{r}
metadata2<-select(metadata, 'Run', 'HIV_serostatus','Host_Age', 'HIV_Profile',  'PCR_human_papilloma_virus', 'host_Residency_Area', 'HCV_coinfection', 'leukocytes','stool_consistency', 'Lymphocytes',  'Anal_cytology')
dim(comparison_df)
```
Add metadata variables to analysis:

```{r}
comparison_df<-merge(comparison_df, metadata2, by.x = 'Sample', by.y = 'Run')
comparison_df2<-merge(comparison_df2, metadata2, by.x = 'Sample', by.y = 'Run')
```

Transform to list grouped by AMR gene family and Drug Class respectively: 
```{r}
comparison_list_amr = split(comparison_df, f = comparison_df$`AMR Gene Family`)
comparison_list_drug = split(comparison_df2, f = comparison_df2$`Drug Class`)
```

Compare RPKM values according to HIV serostatus with the 3 pipelines:   
+Obtain wicox.test p values  
+Merge datsets  

```{r warning=FALSE}
HIVfun<-function(list){
pvalues_RGI<-lapply(list, function(x) wilcox.test(sum_RPKM_RGI~HIV_serostatus, data=x)$p.value)
pvalues_ariba<-lapply(list, function(x) wilcox.test(sum_RPKM_ariba~HIV_serostatus, data=x)$p.value)
pvalues_groot<-lapply(list, function(x) wilcox.test(sum_RPKM_groot~HIV_serostatus, data=x)$p.value)
x<-do.call(rbind.data.frame, pvalues_RGI)
y<-do.call(rbind.data.frame, pvalues_ariba)
z<-do.call(rbind.data.frame, pvalues_groot)
pvalues_all<-cbind(x,y,z)
rownames(pvalues_all)<-names(pvalues_RGI)
names(pvalues_all)<-c("RGI", "Ariba","Groot")
return(pvalues_all)}
```

```{r warning=FALSE}
pvalues_all_amr<-HIVfun(comparison_list_amr)
pvalues_all_drug<-HIVfun(comparison_list_drug)
```

Retain only statistically significant AMR gene families and Drug classes(VIH +-) according to at least one pipeline:

```{r}
library(knitr)
subset_sig<-function(pvalues)
{sig<-subset(pvalues, pvalues$RGI<=0.05 | pvalues$Ariba<=0.05 | pvalues$Groot<=0.05)
  return(sig)}

pvalues_amr_sig<-subset_sig(pvalues_all_amr)
pvalues_drug_sig<-subset_sig(pvalues_all_drug)

kable(round(pvalues_amr_sig,3))
kable(round(pvalues_drug_sig,3))
```

```{r}
##write.csv(pvalues_amr_sig, "pvalues_amr_sig.csv")
##write.csv(pvalues_drug_sig, "pvalues_drug_sig.csv")
```

