---
title: "RPKM"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "~/Documentos/Repos/UOC-Master/testOutputs")
```

## How to calculate RPKM (Reads per Kilobase per Milion)

1. Count up the total reads in a sample and divide that number by 1,000,000 – this is our “per million” scaling factor.
2. Divide the read counts by the “per million” scaling factor. This normalizes for sequencing depth, giving you reads per million (RPM)
3. Divide the RPM values by the length of the gene, in kilobases. This gives you RPKM.

RPKM =   numReads / ( geneLength/1000 * totalNumReads/1,000,000 )

numReads - number of reads mapped to a gene sequence (All Mapped Reads)  
geneLength - length of the gene sequence (Referece Length)  
totalNumReads - total number of mapped reads of a sample (mapping stats)  

# RGI

```{r}
library(readr)
gene_mapping_data <- read_delim("RGI/RTC_149_RGI_output.gene_mapping_data.txt", "\t", escape_double = FALSE, trim_ws = TRUE)
allele_mapping_data <- read_delim("RGI/RTC_149_RGI_output.allele_mapping_data.txt", "\t", escape_double = FALSE, trim_ws = TRUE)
```

```{r}
library(dplyr)
totalMapReads<-19968 ##value obtained from *overall_mapping_stats.txt
totalNumReads<-	17072992

allele_mapping_data<-allele_mapping_data%>%mutate(RPKM=`All Mapped Reads` /(`Reference Length`/1000 * totalNumReads/1000000))
gene_mapping_data<-gene_mapping_data%>%mutate(RPKM=`All Mapped Reads` /(`Reference Length`/1000 * totalNumReads/1000000))
```

### Average MAPQ distribution

```{r}
library(ggplot2)
ggplot(allele_mapping_data, aes(`Average MAPQ (Completely Mapped Reads)`)) +
  geom_histogram(bins=50, aes(y = stat(density)))+geom_density()+theme_bw()

ggplot(allele_mapping_data, aes(`Average MAPQ (Completely Mapped Reads)`)) +
  geom_histogram(bins=50)+theme_bw()

```

###  Apply Average MAPQ filter:

```{r}
allele_mapping_data<-allele_mapping_data%>%filter(`Average MAPQ (Completely Mapped Reads)`>10)
```

### RPKM sum per AMR gene family

```{r}
library(knitr)
n_distinct(allele_mapping_data$`AMR Gene Family`)
AMR_family_RGI<-allele_mapping_data%>%
  group_by(`AMR Gene Family`)%>%
  summarise('number of AMR genes per family (RGI)'=n(),
            sum_RPKM_RGI=sum(RPKM),
            avg_MAPQ=mean(`Average MAPQ (Completely Mapped Reads)`),
            q1_MAPQ=quantile(`Average MAPQ (Completely Mapped Reads)`, 0.25),
            p3_MAPQ=quantile(`Average MAPQ (Completely Mapped Reads)`,0.75),)%>%arrange(desc(sum_RPKM_RGI))
kable(head(AMR_family_RGI))
str(AMR_family_RGI$`sum RPKM (RGI)`)
```


```{r}
ggplot(AMR_family_RGI, aes(log(sum_RPKM_RGI), avg_MAPQ))+geom_point()+theme_bw()
```

# ARIBA

```{r}
report <- read_delim("ARIBA/report.tsv", 
    "\t", escape_double = FALSE, trim_ws = TRUE)
```
Select only coding sequences and remove contig and variant data

```{r}
report<-report%>% filter(gene==1)%>%select(c(1:8,31))%>%unique()
```


ARIBA does not generate the nº of mapped reads, I will use the nº reads per sample (once filtered)

```{r}
totalNumReads<-	17072992
report<-report%>%mutate(RPKM=reads /(ref_len/1000 * totalNumReads/1000000))
```

### Obtain AMR gene family
Import CARD data aro index: 

```{r}
library(readr)
aro_index <- read_delim("aro_index.tsv", 
    "\t", escape_double = FALSE, trim_ws = TRUE)
```

Trasform ARIBA notation to merge with CARD  DB
```{r}
library(stringr)
library(reshape2)
y<-colsplit(report$ref_name, "\\.", c("ARO Term", "ARO Acession","NCBI","init_final","Model Sequence ID","X"))
y$NCBI<-ifelse(str_detect(y$init_final, "_"),y$NCBI ,paste(y$NCBI,".",y$init_final) )
y$init_final<-ifelse(str_detect(y$init_final, "_"), y$init_final, y$`Model Sequence ID`)
y$`Model Sequence ID`<-ifelse(str_detect(y$`Model Sequence ID`, "_"), y$X, y$`Model Sequence ID`)
y<-select(y, -X)
report<-cbind(report,y)

report<-merge(report, select(aro_index,`Model Sequence ID`, `AMR Gene Family`,`Drug Class`, `Resistance Mechanism` ) , by="Model Sequence ID")
str(report$rp)

```

Mapping quality?

### RPKM sum per AMR gene family

```{r}
library(knitr)
library(dplyr)
AMR_family_ARIBA<-report%>%
  group_by(`AMR Gene Family`)%>%
  summarise('number of AMR genes per family(ARIBA)'=n(),
            'sum RPKM (ARIBA)'=sum(RPKM))%>%arrange(desc('sum RPKM (ARIBA)'))

kable(head(AMR_family_ARIBA))
```

# GROOT
Get groot report:

```{r}
library(readr)
groot_report <- read_delim("Groot/filtered_RTC_149-report-0", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)
names(groot_report)<-c("AMR.gene", "read.count", "gene.length", "coverage.cigar")
head(groot_report)
```
Report: This will report gene, read count, gene length, coverage cigar"
Trasform GROOT notation and merge with CARD  DB
```{r}
library(stringr)
library(reshape2)
library(dplyr)
y<-colsplit(groot_report$AMR.gene, "\\.", c("ARO Term", "ARO Acession","NCBI","init_final","Model Sequence ID","X"))

y$NCBI<-ifelse(str_detect(y$init_final, "-"),y$NCBI ,paste(y$NCBI,".",y$init_final) )
y$init_final<-ifelse(str_detect(y$init_final, "-"), y$init_final, y$`Model Sequence ID`)
y$`Model Sequence ID`<-ifelse(str_detect(y$`Model Sequence ID`, "-"), y$X, y$`Model Sequence ID`)
y<-select(y, -X)
groot_report<-cbind(groot_report,y)
groot_report<-merge(groot_report, select(aro_index,`Model Sequence ID`, `AMR Gene Family`,`Drug Class`, `Resistance Mechanism` ) , by="Model Sequence ID")
```

Calculate RPKM:

```{r}
otalNumReads<-	17072992
groot_report<-groot_report%>%mutate(RPKM=read.count /(gene.length/1000 * totalNumReads/1000000))
```

```{r}
library(knitr)
n_distinct(groot_report$`AMR Gene Family`)
AMR_family_Groot<-groot_report%>%
  group_by(`AMR Gene Family`)%>%
  summarise('number of AMR genes per family (groot)'=n(),
            sum_RPKM_groot=sum(RPKM))%>%arrange(desc(sum_RPKM_groot))
kable(head(AMR_family_Groot))
```

# ARIBA vs RGI vs GROOT

```{r}
length(AMR_family_ARIBA$`AMR Gene Family`)
length(AMR_family_RGI$`AMR Gene Family`)
length(AMR_family_Groot$`AMR Gene Family`)
AMR_family<-merge(AMR_family_ARIBA, AMR_family_RGI[ ,1:3], by="AMR Gene Family", all=TRUE)
AMR_family<-merge(AMR_family, AMR_family_Groot, by="AMR Gene Family", all=TRUE)

kable(arrange(AMR_family,`number of AMR genes per family(ARIBA)`))
```

```{r}
##write_tsv(AMR_family, "AMR_family.tsv")
```

